---
title: "R Notebook"
output: NULL
---

## Normalize the path:

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE) 
knitr::opts_knit$set(root.dir=normalizePath('../../')) 
```

#on HCC
```{r}
library(data.table)

#microbiome

geno_name = fread("largedata/Zhikai/BG_genotype_names.csv",header = T, data.table = F)

log_rel_abun_hn = fread("largedata/Zhikai/log_rel_abundance_3626asvs_HN.csv",header = T, data.table = F)
log_rel_abun_ln = fread("largedata/Zhikai/log_rel_abundance_3626asvs_LN.csv",header = T, data.table = F)

rownames(log_rel_abun_hn) = log_rel_abun_hn$ASV

log_rel_abun_hn$ASV = NULL

log_rel_abun_hn_t <- as.data.frame(t(as.matrix(log_rel_abun_hn)))

log_rel_abun_hn_t = cbind(rownames(log_rel_abun_hn_t), log_rel_abun_hn_t)
colnames(log_rel_abun_hn_t)[1] = "Sample_ID"

rownames(log_rel_abun_ln) = log_rel_abun_ln$ASV

log_rel_abun_ln$ASV = NULL

log_rel_abun_ln_t <- as.data.frame(t(as.matrix(log_rel_abun_ln)))

log_rel_abun_ln_t = cbind(rownames(log_rel_abun_ln_t), log_rel_abun_ln_t)
colnames(log_rel_abun_ln_t)[1] = "Sample_ID"

log_rel_abun = rbind(log_rel_abun_hn_t, log_rel_abun_ln_t)

log_rel_1 = log_rel_abun[,1:3]

rm(log_rel_abun_hn)
rm(log_rel_abun_hn_t)
rm(log_rel_abun_ln)
rm(log_rel_abun_ln_t)


sinfo = fread("largedata/Zhikai/sample_data_3626asvs.csv", header = T, data.table = F)
colnames(sinfo)[2] = "MM_name"


```

#on local
```{r}
library(data.table)
library(dplyr)
#genotype
genos = fread("largedata/hmp321_282_agpv4_maf005_miss02_pruned_s50k_imputed_matrix.txt",data.table = F)

geno = genos[,1:3]
rm(genos)

geno_names = merge(geno, geno_name, by.x = "genotype", by.y = "GX_name", all.x = T)

mm_names = unique(geno_names$MM_name)
mm_names2 = unique(sinfo$MM_name)

mm_names2[which(!(mm_names2 %in% mm_names))]

geno_names$MM_name =  ifelse(geno_names$MM_name == "", geno_names$BG_original, geno_names$MM_name)

#sinfo_geno = merge(sinfo, geno_names, by.x = "MM_name", by.y = "MM_name")
#sinfo_geno = sinfo_geno[,-c((ncol(sinfo_geno)-2):ncol(sinfo_geno))]


sinfo_microbiome = merge(sinfo, log_rel_1, by.x = "Sample_ID", by.y = "Sample_ID")
sinfo_microbiome_avg =  as.data.frame( sinfo_microbiome[,c(2:6,8:ncol(sinfo_microbiome))] %>% group_by(row,MM_name, nitrogen, block, sp, spb) %>% summarise_all(mean))

sinfo_microbiome_geno = merge(sinfo_microbiome_avg, geno_names, by.x = "MM_name", by.y = "MM_name")
sinfo_microbiome_geno = sinfo_microbiome_geno[,-c((ncol(sinfo_microbiome_geno)-2):ncol(sinfo_microbiome_geno))]

#phenotype
#vegetation index
vis = fread("data/raw_phe.txt",header = T, data.table = F)
vis = subset(vis, date == "12-Aug")
cobw = fread("data/pheno2019_cob_weight.csv",header = T, data.table = F)
kernelw = fread("data/pheno2019_20_kernel_weight.csv",header = T, data.table = F)


#principal components
pcs = fread("data/hmp321_282_agpv4_maf005_miss03_pruned.eigenvec")
pcs = pcs[,2:12]
colnames(pcs) = c("genotype", paste0(rep("pc",10),1:10))


```

#pheno visualization
```{r}
library(ggplot2)
library(GGally)
ggpairs(pheno[,2:7],title = "Flowering Time Traits Distribution")
```

```{r}
hist(rna_flower$GRMZM2G171650, main = "Gene Expression of MADS69 in Mature Leafe during the Day", xlab = "Fragments Per Million (FPM)", ylab = "Frequency")
```

