---
title: "Genomic Selection practice"
author: "Zhikai Yang"
date: "March 9th, 2022"
output: NULL
---


## Normalize the path:

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE) 
knitr::opts_knit$set(root.dir=normalizePath('../../')) 
```




#3626 asvs 

```{r}
library(data.table)
library(ggplot2)
library(agricolae)
library(ggpubr)
library(dplyr)

pred_all = fread("largedata/review/pred_seeds_vis_shuffled_sample_names_3626asvs_spb_from_count_nlogrel.txt", header = T)


pred_all_avg =  as.data.frame( pred_all %>% group_by(trait, date, method, seed, shuffled) %>% summarise_all(mean))

pred_all_avg$date <- factor(pred_all_avg$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

pred_all_avg = pred_all_avg[-which((pred_all_avg$method == "rrBLUP") & (pred_all_avg$shuffled == "yes")),]

pred_all_avg$method[which((pred_all_avg$method == "rrBLUP_geno_microbiome") & (pred_all_avg$shuffled == "yes"))] = "rrBLUP_geno_shuffled_microbiome"



pred_all_avg$method <- factor(pred_all_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome", "rrBLUP_geno_shuffled_microbiome"), labels =c( "SNP", "SNP + Microbiome", "SNP + shuffled Microbiome"))



vis = as.character(data.frame(table(pred_all_avg$trait))$Var1)


my_comparisons <- list(c("SNP", "SNP + Microbiome"), c("SNP", "SNP + shuffled Microbiome"))





plist = list()

fsize = 16

for (i in 1:8) {
    plist[[i]] = ggboxplot(pred_all_avg[which(pred_all_avg$trait == vis[i]),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    #stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(vis[i]) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(-0.01, 1)) +
    theme(plot.title = element_text(size=16, face = "bold"), 
          axis.text=element_text(size=14, face="bold"),
          strip.text.x = element_text(size = 14, face = "bold"),
          axis.title=element_text(size=14, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=14, face="bold"),
          legend.text = element_text(size=14))

    
}


library(ggpubr)
figure <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]],plist[[5]],plist[[6]],plist[[7]],plist[[8]], ncol = 3, nrow = 3)
figure

pdf("largedata/review/figures/p_vis_spb.pdf", width = 13, height= 10)
figure
dev.off()

i=6

p_vari_shuffled_3626 = ggboxplot(pred_all_avg[which(pred_all_avg$trait == vis[i]),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    xlab("") +
    ylab("") +
    ggtitle("3626 ASVs") +
    theme_classic() +
    labs(fill = "") +
    scale_y_continuous(limits = c(0.4, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))



p_vari_shuffled_aug12_3626 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("") +
    ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


p_vari_shuffled_aug22_3626 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


p_vari_shuffled_sep5_3626 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "bottom", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_test_3626 = ggarrange(p_vari_shuffled_aug12_3626, p_vari_shuffled_aug22_3626, p_vari_shuffled_sep5_3626, nrow = 3, ncol = 1)

p_vari_test_3626



pred_all_compare_mean = as.data.frame( pred_all_avg[,-5] %>% group_by(trait, date, method) %>% summarise_all(mean))
pred_all_compare_mean = pred_all_compare_mean[-which(pred_all_compare_mean$method == "SNP + shuffled Microbiome"),]

rownames(pred_all_compare_mean) = 1:nrow(pred_all_compare_mean)

pred_all_compare_mean_d = pred_all_compare_mean[seq(1,47,by=2),c(1,2,5)]

rownames(pred_all_compare_mean_d) = 1:nrow(pred_all_compare_mean_d)

for (i in seq(1,47,by=2)) {
  pred_all_compare_mean_d$accuracy[((i+1)/2)] = (pred_all_compare_mean$accuracy[i+1]-pred_all_compare_mean$accuracy[i])/pred_all_compare_mean$accuracy[i]
}



```


#154 tax

```{r}

pred_all = fread("largedata/review/pred_seeds_vis_shuffled_154Functional_Tax_Groupes_spb_nlogrel.txt", header = T)


pred_all_avg =  as.data.frame( pred_all %>% group_by(trait, date, method, seed, shuffled) %>% summarise_all(mean))

pred_all_avg$date <- factor(pred_all_avg$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

pred_all_avg = pred_all_avg[-which((pred_all_avg$method == "rrBLUP") & (pred_all_avg$shuffled == "yes")),]

pred_all_avg$method[which((pred_all_avg$method == "rrBLUP_geno_microbiome") & (pred_all_avg$shuffled == "yes"))] = "rrBLUP_geno_shuffled_microbiome"



pred_all_avg$method <- factor(pred_all_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome", "rrBLUP_geno_shuffled_microbiome"), labels =c( "SNP", "SNP + Microbiome", "SNP + shuffled Microbiome"))



vis = as.character(data.frame(table(pred_all_avg$trait))$Var1)


my_comparisons <- list(c("SNP", "SNP + Microbiome"), c("SNP", "SNP + shuffled Microbiome"))



i=6

p_vari_shuffled_154 = ggboxplot(pred_all_avg[which(pred_all_avg$trait == vis[i]),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
  xlab("") +
  ylab("") +
  ggtitle("154 Taxonomic Groups") +
  theme_classic() +
  labs(fill = "") +
  scale_y_continuous(limits = c(0.4, 0.8)) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_text(size=fsize, face="bold"),
        axis.text.x = element_text(angle = 25, hjust=0.8),
        legend.position = "none", 
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))



p_vari_shuffled_aug12_154 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
  xlab("") +
  ylab("Prediction Accuracy ") +
  ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
  theme_classic() +
  labs(fill = "") +
  #scale_y_continuous(limits = c(0.15, 0.8)) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_blank(),
        axis.text.x = element_blank(),
        legend.position = "none", 
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))


p_vari_shuffled_aug22_154 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
  xlab("") +
  ylab("Prediction Accuracy ") +
  ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
  theme_classic() +
  labs(fill = "") +
  #scale_y_continuous(limits = c(0.15, 0.8)) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_blank(),
        axis.text.x = element_blank(),
        legend.position = "none", 
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))


p_vari_shuffled_sep5_154 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
  xlab("") +
  ylab("Prediction Accuracy ") +
  ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
  theme_classic() +
  labs(fill = "") +
  #scale_y_continuous(limits = c(0.15, 0.8)) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_blank(),
        axis.text.x = element_blank(),
        legend.position = "bottom", 
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))

p_vari_test_154 = ggarrange(p_vari_shuffled_aug12_154, p_vari_shuffled_aug22_154, p_vari_shuffled_sep5_154, nrow = 3, ncol = 1)

p_vari_test_154



pred_all_compare_mean = as.data.frame( pred_all_avg[,-5] %>% group_by(trait, date, method) %>% summarise_all(mean))
pred_all_compare_mean = pred_all_compare_mean[-which(pred_all_compare_mean$method == "SNP + shuffled Microbiome"),]

rownames(pred_all_compare_mean) = 1:nrow(pred_all_compare_mean)

pred_all_compare_mean_d = pred_all_compare_mean[seq(1,47,by=2),c(1,2,5)]

rownames(pred_all_compare_mean_d) = 1:nrow(pred_all_compare_mean_d)

for (i in seq(1,47,by=2)) {
  pred_all_compare_mean_d$accuracy[((i+1)/2)] = (pred_all_compare_mean$accuracy[i+1]-pred_all_compare_mean$accuracy[i])/pred_all_compare_mean$accuracy[i]
}


```
# 19 classes

```{r}


pred_all = fread("largedata/review/pred_seeds_vis_shuffled_19classes_spb_nlogrel.txt", header = T)


pred_all_avg =  as.data.frame( pred_all %>% group_by(trait, date, method, seed, shuffled) %>% summarise_all(mean))

pred_all_avg$date <- factor(pred_all_avg$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

pred_all_avg = pred_all_avg[-which((pred_all_avg$method == "rrBLUP") & (pred_all_avg$shuffled == "yes")),]

pred_all_avg$method[which((pred_all_avg$method == "rrBLUP_geno_microbiome") & (pred_all_avg$shuffled == "yes"))] = "rrBLUP_geno_shuffled_microbiome"



pred_all_avg$method <- factor(pred_all_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome", "rrBLUP_geno_shuffled_microbiome"), labels =c( "SNP", "SNP + Microbiome", "SNP + shuffled Microbiome"))



vis = as.character(data.frame(table(pred_all_avg$trait))$Var1)


my_comparisons <- list(c("SNP", "SNP + Microbiome"), c("SNP", "SNP + shuffled Microbiome"))


i=6

p_vari_shuffled_19 = ggboxplot(pred_all_avg[which(pred_all_avg$trait == vis[i]),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
  xlab("") +
  ylab("") +
  ggtitle("19 Classes") +
  theme_classic() +
  labs(fill = "") +
  scale_y_continuous(limits = c(0.4, 0.8)) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_text(size=fsize, face="bold"),
        axis.text.x = element_text(angle = 25, hjust=0.8),
        legend.position = "none", 
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))



p_vari_shuffled_aug12_19 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
  xlab("") +
  ylab("Prediction Accuracy ") +
  ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
  theme_classic() +
  labs(fill = "") +
  #scale_y_continuous(limits = c(0.15, 0.8)) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_blank(),
        axis.text.x = element_blank(),
        legend.position = "none", 
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))


p_vari_shuffled_aug22_19 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
  xlab("") +
  ylab("Prediction Accuracy ") +
  ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
  theme_classic() +
  labs(fill = "") +
  #scale_y_continuous(limits = c(0.15, 0.8)) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_blank(),
        axis.text.x = element_blank(),
        legend.position = "none", 
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))


p_vari_shuffled_sep5_19 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
  xlab("") +
  ylab("Prediction Accuracy ") +
  ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
  theme_classic() +
  labs(fill = "") +
  #scale_y_continuous(limits = c(0.15, 0.8)) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_blank(),
        axis.text.x = element_blank(),
        legend.position = "bottom", 
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))

p_vari_test_19 = ggarrange(p_vari_shuffled_aug12_19, p_vari_shuffled_aug22_19, p_vari_shuffled_sep5_19, nrow = 3, ncol = 1)

p_vari_test_19



pred_all_compare_mean = as.data.frame( pred_all_avg[,-5] %>% group_by(trait, date, method) %>% summarise_all(mean))
pred_all_compare_mean = pred_all_compare_mean[-which(pred_all_compare_mean$method == "SNP + shuffled Microbiome"),]

rownames(pred_all_compare_mean) = 1:nrow(pred_all_compare_mean)

pred_all_compare_mean_d = pred_all_compare_mean[seq(1,47,by=2),c(1,2,5)]

rownames(pred_all_compare_mean_d) = 1:nrow(pred_all_compare_mean_d)

for (i in seq(1,47,by=2)) {
  pred_all_compare_mean_d$accuracy[((i+1)/2)] = (pred_all_compare_mean$accuracy[i+1]-pred_all_compare_mean$accuracy[i])/pred_all_compare_mean$accuracy[i]
}


```

```{r}
p_vari_shuffled = ggarrange(p_vari_shuffled_3626, p_vari_shuffled_154, p_vari_shuffled_19, nrow = 1, ncol = 3)
p_vari_shuffled

pdf("largedata/review/figures/p_vari_shuffled_spb3.pdf", width = 13, height= 6)
p_vari_shuffled
dev.off()


```


# 3626 ASVs shuffled quadrant

```{r}
library(data.table)
library(ggplot2)
library(agricolae)
library(ggpubr)
library(dplyr)

pred_all = fread("largedata/review/pred_seeds_vis_shuffled_sample_names_3626asvs_spb.txt", header = T)


pred_all_avg =  as.data.frame( pred_all %>% group_by(trait, date, method, seed, shuffled) %>% summarise_all(mean))

pred_all_avg$date <- factor(pred_all_avg$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

pred_all_avg = pred_all_avg[-which((pred_all_avg$method == "rrBLUP") & (pred_all_avg$shuffled == "yes")),]

pred_all_avg$method[which((pred_all_avg$method == "rrBLUP_geno_microbiome") & (pred_all_avg$shuffled == "yes"))] = "rrBLUP_geno_shuffled_microbiome"


pred_quadrant = fread("largedata/review/pred_seeds_vis_shuffled_quandrant_yes_3626asvs_spb.txt", header = T)


pred_quadrant_avg =  as.data.frame( pred_quadrant %>% group_by(trait, date, method, seed, shuffled) %>% summarise_all(mean))

pred_quadrant_avg$date <- factor(pred_quadrant_avg$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))


pred_quadrant_avg$method[which((pred_quadrant_avg$method == "rrBLUP_geno_microbiome") & (pred_quadrant_avg$shuffled == "quadrant_yes"))] = "rrBLUP_geno_shuffled_microbiome_quadrant"


pred_all_avg = rbind(pred_all_avg, pred_quadrant_avg)

pred_all_avg$method <- factor(pred_all_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome", "rrBLUP_geno_shuffled_microbiome", "rrBLUP_geno_shuffled_microbiome_quadrant"), labels =c( "SNP", "SNP + ASV", "SNP + shuffled ASV", "SNP + shuffled ASV within quadrant"))



vis = as.character(data.frame(table(pred_all_avg$trait))$Var1)


my_comparisons <- list(c("SNP", "SNP + ASV"), c("SNP", "SNP + shuffled ASV"),  c("SNP + ASV", "SNP + shuffled ASV within quadrant"),   c("SNP + shuffled ASV", "SNP + shuffled ASV within quadrant"))





plist = list()

fsize = 16

for (i in 1:8) {
    plist[[i]] = ggboxplot(pred_all_avg[which(pred_all_avg$trait == vis[i]),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    #stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(vis[i]) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(-0.01, 1)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
figure <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]],plist[[5]],plist[[6]],plist[[7]],plist[[8]], ncol = 2, nrow = 4)
figure


i=6


p_vari_shuffled_aug12 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


p_vari_shuffled_aug22 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


p_vari_shuffled_sep5 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize-5, face="bold"),
          legend.text = element_text(size=fsize-5))

p_vari_shuffled_quadrant = ggarrange(p_vari_shuffled_aug12, p_vari_shuffled_aug22, p_vari_shuffled_sep5, nrow = 1, ncol = 3)

p_vari_shuffled_quadrant


pdf("largedata/review/figures/p_vari_shuffled_quadrant_spb2.pdf", width = 13, height= 6)
p_vari_shuffled_quadrant
dev.off()

```




#coef of microbiome, on HCC cd /common/jyanglab/zhikaiyang/projects/GP_microbiome
```{r}
library(data.table)

f <- list.files(path="largedata/review/output_shuffled_sample_names", pattern="coef_mic_trait.*_spb_jyang", full.names = T)
fn <- list.files(path="largedata/review/output_shuffled_sample_names", pattern="coef_mic_trait.*_nitrogen_.*_spb_jyang", full.names = T)

f = f[-which(f %in% fn)]



out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_spb_jyang.*", "", out$file)
out = out[,-3]



fwrite(out, "largedata/review/coef_mic_all_traits_dates_3626asvs_spb.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```

#coef of microbiome spread
```{r}

coef_all = fread("largedata/review/coef_mic_all_traits_dates_3626asvs_spb.txt", data.table = F)


coef_all$date <- factor(coef_all$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))


vis = as.character(data.frame(table(coef_all$trait))$Var1)

plist = list()

for (i in 1:8) {
    plist[[i]] = ggboxplot(coef_all[which((coef_all$trait == vis[i])),], x = "date", y = "coef", color = "date", palette = "jco",add = "jitter") +
    #stat_compare_means(aes(group = method), label = "p.signif", paired = T)+
    xlab("") +
    ylab("Coef of ASVs ") +
    ggtitle(vis[i]) +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
figure <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]],plist[[5]],plist[[6]],plist[[7]],plist[[8]], ncol = 2, nrow = 4)
figure


coef_all = fread("largedata/review/coef_mic_all_traits_dates_3626asvs_spb.txt", data.table = F)

vis = as.character(data.frame(table(coef_all$trait))$Var1)
date = as.character(data.frame(table(coef_all$date))$Var1)


out <- data.frame()
for(i in 1:length(vis)){
  for (j in 1:length(date)) {
    tem <- coef_all[which((coef_all$trait == vis[i])  & (coef_all$date == date[j])),]
    tem$abs_coef = abs(tem$coef)
    tem = tem[order(-tem$abs_coef),]
    tem$rank = 1:nrow(tem)
    tem$top1p = ifelse(tem$rank <= 36, 1, 0)
    tem$top100 = ifelse(tem$rank <= 100, 1, 0)
    out <- rbind(out, tem)
  }
}

asv_coef_vari = out[which(out$trait == "VARI"),]
asv_coef_vari = asv_coef_vari[which(asv_coef_vari$top1p == 1),]
row.names(asv_coef_vari) = 1:nrow(asv_coef_vari)
id_asv = intersect(intersect(asv_coef_vari$asv[1:36], asv_coef_vari$asv[37:72]),asv_coef_vari$asv[73:108])


asv_mic_group = fread("largedata/elife-75790-supp2-v2.csv", data.table = F)
asv_mic_group = asv_mic_group[,c(1,8)]
asv_coef_vari = merge(asv_coef_vari, asv_mic_group, by.x = "asv", by.y = "ASV")
asv_coef_vari_top1_3dates = asv_coef_vari[which(asv_coef_vari$asv %in% id_asv),]

#intersect(intersect(asv_coef_vari$Functional_Tax_Group[which(asv_coef_vari$date == "Aug12")], asv_coef_vari$Functional_Tax_Group[which(asv_coef_vari$date == "Aug22")]),asv_coef_vari$Functional_Tax_Group[which(asv_coef_vari$date == "Sept5")])
#mic_group_anno = fread("largedata/elife-75790-supp3-v2.csv", data.table = F)


asv_coef_vis = out[which(out$top1p == 1),]
asv_coef_vis = merge(asv_coef_vis, asv_mic_group, by.x = "asv", by.y = "ASV")
asv_coef_vis$nitrogen = "overall"
asv_coef_vis = asv_coef_vis[,c(1:4,10,5:9)]



```



#separate HN and LN, on HCC
```{r}
library(data.table)
f <- list.files(path="largedata/review/output_shuffled_sample_names", pattern="prediction_accuracy_seed.*_nitrogen_.*_spb", full.names = T)

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$nitrogen = gsub(".*nitrogen_|_shuffled.*", "", out$file)
out$shuffled  <- gsub(".*shuffled_|_spb.*", "", out$file)

id_pc = which(out$method == "rrBLUP_PC_microbiome")
out = out[-id_pc,]


fwrite(out[,-7], "largedata/review/pred_seeds_nitrogen_vis_shuffled_names_3626asvs_spb.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```


#separate HN and LN, all vegetation indexes visualization 
```{r}
pred_all_nitrogen = fread("largedata/review/pred_seeds_nitrogen_vis_shuffled_names_3626asvs_spb.txt", data.table = F)

pred_all_nitrogen_avg =  as.data.frame( pred_all_nitrogen %>% group_by(trait, date, method, seed, nitrogen, shuffled) %>% summarise_all(mean))

pred_all_nitrogen_avg$date <- factor(pred_all_nitrogen_avg$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

pred_all_nitrogen_avg = pred_all_nitrogen_avg[-which((pred_all_nitrogen_avg$method == "rrBLUP") & (pred_all_nitrogen_avg$shuffled == "yes")),]

pred_all_nitrogen_avg$method[which((pred_all_nitrogen_avg$method == "rrBLUP_geno_microbiome") & (pred_all_nitrogen_avg$shuffled == "yes"))] = "rrBLUP_geno_shuffled_microbiome"



pred_all_nitrogen_avg$method <- factor(pred_all_nitrogen_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome", "rrBLUP_geno_shuffled_microbiome"), labels =c( "SNP", "SNP + ASV", "SNP + shuffled ASV"))



vis = as.character(data.frame(table(pred_all_nitrogen_avg$trait))$Var1)

i=6

my_comparisons <- list(c("SNP", "SNP + ASV"), c("SNP", "SNP + shuffled ASV"))


p_vari_shuffled_hn = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "HN")),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    #stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle("HN") +
    theme_classic() +
    labs(fill = "") +
    scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_ln = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "LN")),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    #stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle("LN") +
    theme_classic() +
    labs(fill = "") +
    scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_n_c <- ggarrange(p_vari_shuffled_hn, p_vari_shuffled_ln, ncol = 2, nrow = 1)
p_vari_n_c



pdf("largedata/review/figures/p_vari_n_c_spb.pdf", width = 10, height= 6)
p_vari_n_c
dev.off()


p_vari_shuffled_hn_aug12 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "HN") & (pred_all_nitrogen_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_hn_aug22 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "HN") & (pred_all_nitrogen_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_hn_sep5 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "HN") & (pred_all_nitrogen_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_blank(),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


p_vari_shuffled_ln_aug12 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "LN") & (pred_all_nitrogen_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_ln_aug22 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "LN") & (pred_all_nitrogen_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_ln_sep5 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "LN") & (pred_all_nitrogen_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_hn_c <- ggarrange(p_vari_shuffled_hn_aug12, p_vari_shuffled_hn_aug22, p_vari_shuffled_hn_sep5, ncol = 1, nrow = 3)
p_vari_hn_c

p_vari_ln_c <- ggarrange(p_vari_shuffled_ln_aug12, p_vari_shuffled_ln_aug22, p_vari_shuffled_ln_sep5, ncol = 1, nrow = 3)
p_vari_ln_c


```







#coef of microbiome, on HCC cd /common/jyanglab/zhikaiyang/projects/GP_microbiome
```{r}
library(data.table)

f <- list.files(path="largedata/review/output_shuffled_sample_names", pattern="coef_mic_trait.*_nitrogen_.*_spb_jyang", full.names = T)


out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_nitrogen.*", "", out$file)
out$nitrogen = gsub(".*nitrogen_|_spb_jyang.*", "", out$file)

out = out[,-3]



fwrite(out, "largedata/review/coef_mic_all_traits_dates_nitrogen_3626asvs_spb.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```


```{r}
coef_vis = fread("largedata/review/coef_mic_all_traits_dates_3626asvs_spb.txt", data.table = F)

date = as.character(data.frame(table(coef_vis$date))$Var1)

vis = vis = as.character(data.frame(table(coef_vis$trait))$Var1)

i = 6

fsize = 16

p_vari_coef_mic = ggboxplot(coef_vis[which((coef_vis$trait == vis[i])),], x = "date", y = "coef", color = "date", palette = "jco",add = "jitter") +
    xlab("") +
    ylab("Coef of ASVs ") +
    ggtitle(vis[i]) +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_coef_mic
```
#coef of microbiome under separate N, and output top 1% ASV for overall, HN, LN
```{r}
coef_vis = fread("largedata/review/coef_mic_all_traits_dates_nitrogen_3626asvs_spb.txt", data.table = F)

date = as.character(data.frame(table(coef_vis$date))$Var1)
vis = as.character(data.frame(table(coef_vis$trait))$Var1)



coef_vis_hn = coef_vis[which(coef_vis$nitrogen == "HN"),]
out <- data.frame()
for(i in 1:length(vis)){
  for (j in 1:length(date)) {
    tem <- coef_vis_hn[which((coef_vis_hn$trait == vis[i])  & (coef_vis_hn$date == date[j])),]
    tem$abs_coef = abs(tem$coef)
    tem = tem[order(-tem$abs_coef),]
    tem$rank = 1:nrow(tem)
    tem$top1p = ifelse(tem$rank <= 36, 1, 0)
    tem$top100 = ifelse(tem$rank <= 100, 1, 0)
    out <- rbind(out, tem)
  }
}

asv_coef_vari_hn = out[which(out$trait == "VARI"),]
asv_coef_vari_hn = asv_coef_vari_hn[which(asv_coef_vari_hn$top1p == 1),]
row.names(asv_coef_vari_hn) = 1:nrow(asv_coef_vari_hn)
id_asv_hn = intersect(intersect(asv_coef_vari_hn$asv[1:36], asv_coef_vari_hn$asv[37:72]),asv_coef_vari_hn$asv[73:108])


asv_mic_group = fread("largedata/elife-75790-supp2-v2.csv", data.table = F)
asv_mic_group = asv_mic_group[,c(1,8)]
asv_coef_vari_hn = merge(asv_coef_vari_hn, asv_mic_group, by.x = "asv", by.y = "ASV")
asv_coef_vari_top1_3dates_hn = asv_coef_vari_hn[which(asv_coef_vari_hn$asv %in% id_asv_hn),]


asv_coef_vis_hn = out[which(out$top1p == 1),]
asv_coef_vis_hn = merge(asv_coef_vis_hn, asv_mic_group, by.x = "asv", by.y = "ASV")




#asv_coef_vis_hn = as.data.frame(out[,-(3:5)] %>% group_by(asv) %>% summarise_all(sum))
#asv_date_coef_vis_hn = as.data.frame(out[,-c(3,5)] %>% group_by(date,asv) %>% summarise_all(sum))






coef_vis_ln = coef_vis[which(coef_vis$nitrogen == "LN"),]
out <- data.frame()
for(i in 1:length(vis)){
  for (j in 1:length(date)) {
    tem <- coef_vis_ln[which((coef_vis_ln$trait == vis[i])  & (coef_vis_ln$date == date[j])),]
    tem$abs_coef = abs(tem$coef)
    tem = tem[order(-tem$abs_coef),]
    tem$rank = 1:nrow(tem)
    tem$top1p = ifelse(tem$rank <= 36, 1, 0)
    tem$top100 = ifelse(tem$rank <= 100, 1, 0)
    out <- rbind(out, tem)
  }
}

asv_coef_vari_ln = out[which(out$trait == "VARI"),]
asv_coef_vari_ln = asv_coef_vari_ln[which(asv_coef_vari_ln$top1p == 1),]
row.names(asv_coef_vari_ln) = 1:nrow(asv_coef_vari_ln)
id_asv_ln = intersect(intersect(asv_coef_vari_ln$asv[1:36], asv_coef_vari_ln$asv[37:72]),asv_coef_vari_ln$asv[73:108])


asv_mic_group = fread("largedata/elife-75790-supp2-v2.csv", data.table = F)
asv_mic_group = asv_mic_group[,c(1,8)]
asv_coef_vari_ln = merge(asv_coef_vari_ln, asv_mic_group, by.x = "asv", by.y = "ASV")
asv_coef_vari_top1_3dates_ln = asv_coef_vari_ln[which(asv_coef_vari_ln$asv %in% id_asv_ln),]


asv_coef_vis_ln = out[which(out$top1p == 1),]
asv_coef_vis_ln = merge(asv_coef_vis_ln, asv_mic_group, by.x = "asv", by.y = "ASV")




#asv_coef_vis_ln = as.data.frame(out[,-(3:5)] %>% group_by(asv) %>% summarise_all(sum))
#asv_date_coef_vis_ln = as.data.frame(out[,-c(3,5)] %>% group_by(date,asv) %>% summarise_all(sum))





fwrite(rbind(asv_coef_vis, asv_coef_vis_hn, asv_coef_vis_ln), "data/supplementary/Supplemental_Table_S1_top_one_percent_asvs.txt", sep="\t", row.names = F, quote = F)

```

#overlap top1% ASV with mediators for VARI
```{r}
mediators_asvs = fread("largedata/review/mediators_asvs_overall_3626asvs_spb.txt")

mediators_asvs_sig = mediators_asvs[which(mediators_asvs$padj<=0.05),]

asv_mic_group = fread("largedata/elife-75790-supp2-v2.csv", data.table = F)
asv_mic_group = asv_mic_group[,c(1,8)]

mediators_asvs_sig_group = merge(asv_mic_group, mediators_asvs_sig,  by.x = "ASV", by.y = "id")
mediators_asvs_sig_group_vari = mediators_asvs_sig_group[which(mediators_asvs_sig_group$trait == "VARI"),]
#fwrite(mediators_asvs_sig_group, "data/supplementary/Supplemental_Table_S2_mediator_asvs.txt", sep="\t", row.names = F, quote = F)

id_asv_med = intersect(mediators_asvs_sig_group_vari$ASV, asv_coef_vari$asv)
id_asv_med_hn = intersect(mediators_asvs_sig_group_vari$ASV, asv_coef_vari_hn$asv)
id_asv_med_ln = intersect(mediators_asvs_sig_group_vari$ASV, asv_coef_vari_ln$asv)
mediators_asvs_sig_group_vari_top1 = mediators_asvs_sig_group_vari[which(mediators_asvs_sig_group_vari$ASV %in% id_asv_med),]
```

#top abs_scaled asvs
```{r}
coef_vis = fread("largedata/review/coef_mic_all_traits_dates_nitrogen_3626asvs_spb.txt", data.table = F)

date = as.character(data.frame(table(coef_vis$date))$Var1)
vis = as.character(data.frame(table(coef_vis$trait))$Var1)



coef_vis_hn = coef_vis[which(coef_vis$nitrogen == "HN"),]
out <- data.frame()
for(i in 1:length(vis)){
  for (j in 1:length(date)) {
    tem <- coef_vis_hn[which((coef_vis_hn$trait == vis[i])  & (coef_vis_hn$date == date[j])),]
    tem$abs_coef = abs(tem$coef)
    tem = tem[order(-tem$abs_coef),]
    tem$rank = 1:nrow(tem)
    tem$top1p = ifelse(tem$rank <= 36, 1, 0)
    tem$top100 = ifelse(tem$rank <= 100, 1, 0)
    out <- rbind(out, tem)
  }
}

asv_coef_vis_hn = as.data.frame(out[,-(3:5)] %>% group_by(asv) %>% summarise_all(sum))
asv_date_coef_vis_hn = as.data.frame(out[,-c(3,5)] %>% group_by(date,asv) %>% summarise_all(sum))

date_asv_top1_hn = asv_date_coef_vis_hn[,c(1,2,6)]
asv_mic_group = fread("largedata/elife-75790-supp2-v2.csv", data.table = F)
asv_mic_group = asv_mic_group[,c(1,8)]
date_asv_mic_group_top1_hn = merge(date_asv_top1_hn, asv_mic_group, by.x = "asv", by.y = "ASV")


coef_vis_ln = coef_vis[which(coef_vis$nitrogen == "LN"),]
out <- data.frame()
for(i in 1:length(vis)){
  for (j in 1:length(date)) {
    tem <- coef_vis_ln[which((coef_vis_ln$trait == vis[i])  & (coef_vis_ln$date == date[j])),]
    tem$abs_coef = abs(tem$coef)
    tem = tem[order(-tem$abs_coef),]
    tem$rank = 1:nrow(tem)
    tem$top1p = ifelse(tem$rank <= 36, 1, 0)
    tem$top100 = ifelse(tem$rank <= 100, 1, 0)
    out <- rbind(out, tem)
  }
}

asv_coef_vis_ln = as.data.frame(out[,-(3:5)] %>% group_by(asv) %>% summarise_all(sum))
asv_date_coef_vis_ln = as.data.frame(out[,-c(3,5)] %>% group_by(date,asv) %>% summarise_all(sum))

date_asv_top1_ln = asv_date_coef_vis_ln[,c(1,2,6)]
asv_mic_group = fread("largedata/elife-75790-supp2-v2.csv", data.table = F)
asv_mic_group = asv_mic_group[,c(1,8)]
date_asv_mic_group_top1_ln = merge(date_asv_top1_ln, asv_mic_group, by.x = "asv", by.y = "ASV")


mic_group_anno = fread("largedata/elife-75790-supp3-v2.csv", data.table = F)


#HN
date_asv_mic_group_top1_anno_HN = merge(date_asv_mic_group_top1_hn, mic_group_anno[which(mic_group_anno$nitrogen_treatment == "HN"),], by.x = "Functional_Tax_Group", by.y ="Microbial_Group")

date_asv_mic_group_top1_anno_HN$iftop1p = ifelse(date_asv_mic_group_top1_anno_HN$top1p >= 1, "Top 1% ASVs", "Not Top 1% ASVs")

date_asv_mic_group_top1_anno_HN$date <- factor(date_asv_mic_group_top1_anno_HN$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

#LN
date_asv_mic_group_top1_anno_LN = merge(date_asv_mic_group_top1_ln, mic_group_anno[which(mic_group_anno$nitrogen_treatment == "LN"),], by.x = "Functional_Tax_Group", by.y ="Microbial_Group")

date_asv_mic_group_top1_anno_LN$iftop1p = ifelse(date_asv_mic_group_top1_anno_LN$top1p >= 1, "Top 1% ASVs", "Not Top 1% ASVs")

date_asv_mic_group_top1_anno_LN$date <- factor(date_asv_mic_group_top1_anno_LN$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

#date_asv_mic_group_jtop1_anno_LN_aug12 = date_asv_mic_group_top1_anno_LN[which((date_asv_mic_group_top1_anno_LN$date == "Aug-12") & (date_asv_mic_group_top1_anno_LN$iftop1p == "Top 1% ASVs")), ]

#fwrite(date_asv_mic_group_jtop1_anno_LN_aug12, "largedata/review/date_asv_mic_group_jtop1_anno_LN_aug12_spb.txt", sep = "\t", quote = F, row.names = F, col.names = T)

#date_asv_mic_group_jtop1_anno_LN_aug12 = fread("largedata/review/date_asv_mic_group_jtop1_anno_LN_aug12_spb.txt", data.table = F)

p_h2_HN = ggboxplot(date_asv_mic_group_top1_anno_HN, x = "date", y = "Heritability", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F, method = "wilcox.test")+
    xlab("") +
    ylab("Heritability") +
    ggtitle("HN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_h2_HN

wilcox.test(date_asv_mic_group_top1_anno_HN$Heritability[which((date_asv_mic_group_top1_anno_HN$top1p == 0) & (date_asv_mic_group_top1_anno_HN$date == "Aug-12"))], date_asv_mic_group_top1_anno_HN$Heritability[which((!(date_asv_mic_group_top1_anno_HN$top1p == 0)) & (date_asv_mic_group_top1_anno_HN$date == "Aug-12"))])

wilcox.test(date_asv_mic_group_top1_anno_HN$Heritability[which((date_asv_mic_group_top1_anno_HN$top1p == 0) & (date_asv_mic_group_top1_anno_HN$date == "Aug-22"))], date_asv_mic_group_top1_anno_HN$Heritability[which((!(date_asv_mic_group_top1_anno_HN$top1p == 0)) & (date_asv_mic_group_top1_anno_HN$date == "Aug-22"))])

wilcox.test(date_asv_mic_group_top1_anno_HN$Heritability[which((date_asv_mic_group_top1_anno_HN$top1p == 0) & (date_asv_mic_group_top1_anno_HN$date == "Sep-5"))], date_asv_mic_group_top1_anno_HN$Heritability[which((!(date_asv_mic_group_top1_anno_HN$top1p == 0)) & (date_asv_mic_group_top1_anno_HN$date == "Sep-5"))])


p_h2_LN = ggboxplot(date_asv_mic_group_top1_anno_LN, x = "date", y = "Heritability", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F, method = "wilcox.test")+
    xlab("") +
    ylab("Heritability") +
    ggtitle("LN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_h2_LN

wilcox.test(date_asv_mic_group_top1_anno_LN$Heritability[which((date_asv_mic_group_top1_anno_LN$top1p == 0) & (date_asv_mic_group_top1_anno_LN$date == "Aug-12"))], date_asv_mic_group_top1_anno_LN$Heritability[which((!(date_asv_mic_group_top1_anno_LN$top1p == 0)) & (date_asv_mic_group_top1_anno_LN$date == "Aug-12"))])

wilcox.test(date_asv_mic_group_top1_anno_LN$Heritability[which((date_asv_mic_group_top1_anno_LN$top1p == 0) & (date_asv_mic_group_top1_anno_LN$date == "Aug-22"))], date_asv_mic_group_top1_anno_LN$Heritability[which((!(date_asv_mic_group_top1_anno_LN$top1p == 0)) & (date_asv_mic_group_top1_anno_LN$date == "Aug-22"))])

wilcox.test(date_asv_mic_group_top1_anno_LN$Heritability[which((date_asv_mic_group_top1_anno_LN$top1p == 0) & (date_asv_mic_group_top1_anno_LN$date == "Sep-5"))], date_asv_mic_group_top1_anno_LN$Heritability[which((!(date_asv_mic_group_top1_anno_LN$top1p == 0)) & (date_asv_mic_group_top1_anno_LN$date == "Sep-5"))])


p_Selection_coeff_HN = ggboxplot(date_asv_mic_group_top1_anno_HN, x = "date", y = "Selection_coeff", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F, method = "wilcox.test")+
    xlab("") +
    ylab("Selection_coeff") +
    ggtitle("HN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_Selection_coeff_HN

p_Selection_coeff_LN = ggboxplot(date_asv_mic_group_top1_anno_LN, x = "date", y = "Selection_coeff", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F, method = "wilcox.test")+
    xlab("") +
    ylab("Selection_coeff") +
    ggtitle("LN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_Selection_coeff_LN

library(ggpubr)
p_anno <- ggarrange(p_h2_HN, p_h2_LN, p_Selection_coeff_HN, p_Selection_coeff_LN, ncol = 2, nrow = 2)
p_anno

pdf("largedata/review/figures/p_anno_spb.pdf", width = 12, height= 12)
p_anno
dev.off()

p_Selection_coeff_LN = ggboxplot(date_asv_mic_group_top1_anno_LN, x = "date", y = "Selection_coeff", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F)+
    xlab("") +
    ylab("Selection_coeff") +
    ggtitle("LN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "bottom", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_Selection_coeff_LN
pdf("largedata/p_Selection_coeff_LN.pdf", width = 12, height= 12)
p_Selection_coeff_LN
dev.off()

```


#coef of mic vs h2 of mic
```{r}
coef_vis = fread("largedata/review/coef_mic_all_traits_dates_nitrogen_3626asvs_spb.txt", data.table = F)

date = as.character(data.frame(table(coef_vis$date))$Var1)
vis = as.character(data.frame(table(coef_vis$trait))$Var1)

coef_vis_hn = coef_vis[which(coef_vis$nitrogen == "HN"),]
out <- data.frame()
for(i in 1:length(vis)){
  for (j in 1:length(date)) {
    tem <- coef_vis_hn[which((coef_vis_hn$trait == vis[i])  & (coef_vis_hn$date == date[j])),]
    tem$abs_coef = abs(tem$coef)
    tem = tem[order(-tem$abs_coef),]
    tem$rank = 1:nrow(tem)
    tem$top1p = ifelse(tem$rank <= 36, 1, 0)
    tem$top100 = ifelse(tem$rank <= 100, 1, 0)
    out <- rbind(out, tem)
  }
}

asv_date_coef_vari_hn = out[which(out$trait == "VARI"), ]



coef_vis_ln = coef_vis[which(coef_vis$nitrogen == "LN"),]
out <- data.frame()
for(i in 1:length(vis)){
  for (j in 1:length(date)) {
    tem <- coef_vis_ln[which((coef_vis_ln$trait == vis[i])  & (coef_vis_ln$date == date[j])),]
    tem$abs_coef = abs(tem$coef)
    tem = tem[order(-tem$abs_coef),]
    tem$rank = 1:nrow(tem)
    tem$top1p = ifelse(tem$rank <= 36, 1, 0)
    tem$top100 = ifelse(tem$rank <= 100, 1, 0)
    out <- rbind(out, tem)
  }
}

asv_date_coef_vari_ln = out[which(out$trait == "VARI"), ]





asv_mic_group = fread("largedata/elife-75790-supp2-v2.csv", data.table = F)
asv_mic_group = asv_mic_group[,c(1,8)]
date_asv_mic_group_top1_ln = merge(asv_date_coef_vari_ln, asv_mic_group, by.x = "asv", by.y = "ASV")
date_asv_mic_group_top1_hn = merge(asv_date_coef_vari_hn, asv_mic_group, by.x = "asv", by.y = "ASV")


mic_group_anno = fread("largedata/elife-75790-supp3-v2.csv", data.table = F)


#HN
date_asv_mic_group_top1_anno_HN = merge(date_asv_mic_group_top1_hn, mic_group_anno[which(mic_group_anno$nitrogen_treatment == "HN"),], by.x = "Functional_Tax_Group", by.y ="Microbial_Group")


#date_asv_mic_group_top1_anno_HN$date <- factor(date_asv_mic_group_top1_anno_HN$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

cor.test(date_asv_mic_group_top1_anno_HN$abs_coef[which(date_asv_mic_group_top1_anno_HN$date == "Aug12")], date_asv_mic_group_top1_anno_HN$Heritability[which(date_asv_mic_group_top1_anno_HN$date == "Aug12")], method = "spearman")

cor.test(date_asv_mic_group_top1_anno_HN$abs_coef[which(date_asv_mic_group_top1_anno_HN$date == "Aug22")], date_asv_mic_group_top1_anno_HN$Heritability[which(date_asv_mic_group_top1_anno_HN$date == "Aug22")], method = "spearman")

cor.test(date_asv_mic_group_top1_anno_HN$abs_coef[which(date_asv_mic_group_top1_anno_HN$date == "Sept5")], date_asv_mic_group_top1_anno_HN$Heritability[which(date_asv_mic_group_top1_anno_HN$date == "Sept5")], method = "spearman")


coef_h2_hn <- ggplot(date_asv_mic_group_top1_anno_HN, aes(x=abs_coef, y=Heritability, color=date)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "bottom",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
coef_h2_hn



#LN
date_asv_mic_group_top1_anno_LN = merge(date_asv_mic_group_top1_ln, mic_group_anno[which(mic_group_anno$nitrogen_treatment == "LN"),], by.x = "Functional_Tax_Group", by.y ="Microbial_Group")


#date_asv_mic_group_top1_anno_LN$date <- factor(date_asv_mic_group_top1_anno_LN$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

cor.test(date_asv_mic_group_top1_anno_LN$abs_coef[which(date_asv_mic_group_top1_anno_LN$date == "Aug12")], date_asv_mic_group_top1_anno_LN$Heritability[which(date_asv_mic_group_top1_anno_LN$date == "Aug12")], method = "spearman")

cor.test(date_asv_mic_group_top1_anno_LN$abs_coef[which(date_asv_mic_group_top1_anno_LN$date == "Aug22")], date_asv_mic_group_top1_anno_LN$Heritability[which(date_asv_mic_group_top1_anno_LN$date == "Aug22")], method = "spearman")

cor.test(date_asv_mic_group_top1_anno_LN$abs_coef[which(date_asv_mic_group_top1_anno_LN$date == "Sept5")], date_asv_mic_group_top1_anno_LN$Heritability[which(date_asv_mic_group_top1_anno_LN$date == "Sept5")], method = "spearman")

coef_h2_ln <- ggplot(date_asv_mic_group_top1_anno_LN, aes(x=abs_coef, y=Heritability, color=date)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "bottom",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
coef_h2_ln




```
#if certain tax enriched with large effect
```{r}
kruskal.test(date_asv_mic_group_top1_anno_HN$abs_coef[which(date_asv_mic_group_top1_anno_HN$date == "Aug12")], date_asv_mic_group_top1_anno_HN$Functional_Tax_Group[which(date_asv_mic_group_top1_anno_HN$date == "Aug12")])

kruskal.test(date_asv_mic_group_top1_anno_HN$abs_coef[which(date_asv_mic_group_top1_anno_HN$date == "Aug22")], date_asv_mic_group_top1_anno_HN$Functional_Tax_Group[which(date_asv_mic_group_top1_anno_HN$date == "Aug22")])

kruskal.test(date_asv_mic_group_top1_anno_HN$abs_coef[which(date_asv_mic_group_top1_anno_HN$date == "Sept5")], date_asv_mic_group_top1_anno_HN$Functional_Tax_Group[which(date_asv_mic_group_top1_anno_HN$date == "Sept5")])


kruskal.test(date_asv_mic_group_top1_anno_LN$abs_coef[which(date_asv_mic_group_top1_anno_LN$date == "Aug12")], date_asv_mic_group_top1_anno_LN$Functional_Tax_Group[which(date_asv_mic_group_top1_anno_LN$date == "Aug12")])

kruskal.test(date_asv_mic_group_top1_anno_LN$abs_coef[which(date_asv_mic_group_top1_anno_LN$date == "Aug22")], date_asv_mic_group_top1_anno_LN$Functional_Tax_Group[which(date_asv_mic_group_top1_anno_LN$date == "Aug22")])

kruskal.test(date_asv_mic_group_top1_anno_LN$abs_coef[which(date_asv_mic_group_top1_anno_LN$date == "Sept5")], date_asv_mic_group_top1_anno_LN$Functional_Tax_Group[which(date_asv_mic_group_top1_anno_LN$date == "Sept5")])


date_asv_mic_group_top1_anno_LN_mean = as.data.frame(date_asv_mic_group_top1_anno_LN[,c(1,3,5,7:10)] %>% group_by(date,Functional_Tax_Group) %>% summarise_all(mean))

date_asv_mic_group_top1_anno_HN_mean = as.data.frame(date_asv_mic_group_top1_anno_HN[,c(1,3,5,7:10)] %>% group_by(date,Functional_Tax_Group) %>% summarise_all(mean))
```




#coef of snps, on HCC cd /common/jyanglab/zhikaiyang/projects/GP_microbiome
```{r}
library(data.table)

f <- list.files(path="largedata/review/output_shuffled_sample_names", pattern="coef_snps_trait.*_spb_jyang", full.names = T)
fn <- list.files(path="largedata/review/output_shuffled_sample_names", pattern="coef_snps_trait.*_nitrogen_.*_spb_jyang", full.names = T)

f = f[-which(f %in% fn)]



out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_spb_jyang.*", "", out$file)
out = out[,-3]



fwrite(out, "largedata/review/coef_snps_all_traits_dates_3626asvs_spb.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```


#coef of snps
```{r}

coef_all_snps = fread("largedata/review/coef_snps_all_traits_dates_3626asvs_spb.txt", data.table = F)

coef_all_snps$date <- factor(coef_all_snps$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))


vis = as.character(data.frame(table(coef_all_snps$trait))$Var1)
i = 6

fsize = 16

p_vari_coef_snps = ggboxplot(coef_all_snps[which((coef_all_snps$trait == vis[i])),][sample(1:150000,15000),], x = "date", y = "coef", color = "date", palette = "jco",add = "jitter") +
    xlab("") +
    ylab("Coef of SNPs ") +
    ggtitle(vis[i]) +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_coef_snps

figure3 <- ggarrange(p_vari_coef_mic, p_vari_coef_snps, ncol = 2, nrow = 1)




pdf("largedata/review/figures/p_vari_coef_mic_snps_spb.pdf", width = 10, height= 6)
figure3
dev.off()


```


# kinblup, separate HN and LN, all vegetation indexes visualization 
```{r}
pred_all_nitrogen = fread("largedata/review/pred_seeds_nitrogen_vis_shuffled_names_154tax_kinblup_spb.txt", data.table = F)

pred_all_nitrogen_avg =  as.data.frame( pred_all_nitrogen %>% group_by(trait, date, method, seed, nitrogen, shuffled) %>% summarise_all(mean))

pred_all_nitrogen_avg$date <- factor(pred_all_nitrogen_avg$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

pred_all_nitrogen_avg = pred_all_nitrogen_avg[-which((pred_all_nitrogen_avg$method == "rrBLUP") & (pred_all_nitrogen_avg$shuffled == "yes")),]

pred_all_nitrogen_avg = pred_all_nitrogen_avg[-which((pred_all_nitrogen_avg$method == "rrBLUP_geno_microbiome") & (pred_all_nitrogen_avg$shuffled == "yes")),]


pred_all_nitrogen_avg = pred_all_nitrogen_avg[-which(pred_all_nitrogen_avg$shuffled == "snps_yes"),]


pred_all_nitrogen_avg$method <- factor(pred_all_nitrogen_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome"), labels =c( "SNP", "SNP + Microbiome"))



vis = as.character(data.frame(table(pred_all_nitrogen_avg$trait))$Var1)

i=6

my_comparisons <- c("SNP", "SNP + Microbiome")


p_vari_shuffled_hn = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "HN")),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    #stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle("HN") +
    theme_classic() +
    labs(fill = "") +
    scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_ln = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "LN")),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    #stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle("LN") +
    theme_classic() +
    labs(fill = "") +
    scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_n_c <- ggarrange(p_vari_shuffled_hn, p_vari_shuffled_ln, ncol = 2, nrow = 1)
p_vari_n_c



pdf("largedata/review/figures/p_vari_n_c_kinblup_spb.pdf", width = 10, height= 6)
p_vari_n_c
dev.off()


p_vari_shuffled_hn_aug12 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "HN") & (pred_all_nitrogen_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(label = "p.signif", method = "t.test", aes(group = method), paired = F,method.args = list(alternative = "greater"))+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_hn_aug22 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "HN") & (pred_all_nitrogen_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(label = "p.signif", method = "t.test", aes(group = method), paired = F, method.args = list(alternative = "greater"))+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_hn_sep5 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "HN") & (pred_all_nitrogen_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(label = "p.signif", method = "t.test", aes(group = method), paired = F, method.args = list(alternative = "greater"))+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_blank(),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


p_vari_shuffled_ln_aug12 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "LN") & (pred_all_nitrogen_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(label = "p.signif", method = "t.test", aes(group = method), paired = F, method.args = list(alternative = "greater"))+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_ln_aug22 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "LN") & (pred_all_nitrogen_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(label = "p.signif", method = "t.test", aes(group = method), paired = F, method.args = list(alternative = "greater"))+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_ln_sep5 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "LN") & (pred_all_nitrogen_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(label = "p.signif", method = "t.test", aes(group = method), paired = F,method.args = list(alternative = "greater"))+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_hn_c <- ggarrange(p_vari_shuffled_hn_aug12, p_vari_shuffled_hn_aug22, p_vari_shuffled_hn_sep5, ncol = 1, nrow = 3)
p_vari_hn_c

p_vari_ln_c <- ggarrange(p_vari_shuffled_ln_aug12, p_vari_shuffled_ln_aug22, p_vari_shuffled_ln_sep5, ncol = 1, nrow = 3)
p_vari_ln_c


```
#yieldc
```{r}
library(data.table)
library(ggplot2)
library(agricolae)
library(ggpubr)
library(dplyr)

pred_yieldc = fread("largedata/review/pred_seeds_yieldc_shuffled_sample_names_3626asvs_spb.txt", data.table = F)

pred_yieldc_avg =  as.data.frame( pred_yieldc %>% group_by(trait, method, seed, shuffled) %>% summarise_all(mean))

pred_yieldc_avg = pred_yieldc_avg[-which((pred_yieldc_avg$method == "rrBLUP") & (pred_yieldc_avg$shuffled == "yes")),]

pred_yieldc_avg$method[which((pred_yieldc_avg$method == "rrBLUP_geno_microbiome") & (pred_yieldc_avg$shuffled == "yes"))] = "rrBLUP_geno_shuffled_microbiome"



pred_yieldc_avg$method <- factor(pred_yieldc_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome", "rrBLUP_geno_shuffled_microbiome"), labels =c( "SNP", "SNP + Microbiome", "SNP + shuffled Microbiome"))





yieldc = as.character(data.frame(table(pred_yieldc_avg$trait))$Var1)

fsize = 16
i = 2

my_comparisons <- list(c("SNP", "SNP + Microbiome"), c("SNP", "SNP + shuffled Microbiome"))

compare_means(accuracy ~ method, pred_yieldc_avg[which( (pred_yieldc_avg$trait == yieldc[i])),], method = "t.test", paired = F)



p_cobw_shuffled = ggboxplot(pred_yieldc_avg[which((pred_yieldc_avg$trait == yieldc[i]) ),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(yieldc[i]) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_cobw_shuffled

plist = list()

fsize = 16

for (i in 1:4) {
    plist[[i]] = ggboxplot(pred_yieldc_avg[which(pred_yieldc_avg$trait == yieldc[i]),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("") +
    ggtitle(yieldc[i]) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x=element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
figure <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]], ncol = 2, nrow = 2)
figure


```


#yieldc separate HN and LN
```{r}
pred_all_nitrogen = fread("largedata/review/pred_seeds_yieldc_nitrogen_shuffled_sample_names_3626asvs_spb.txt", data.table = F)
colnames(pred_all_nitrogen)[2] = "trait"
pred_all_nitrogen_avg =  as.data.frame( pred_all_nitrogen %>% group_by(trait, method, seed, nitrogen, shuffled) %>% summarise_all(mean))

pred_all_nitrogen_avg = pred_all_nitrogen_avg[-which((pred_all_nitrogen_avg$method == "rrBLUP") & (pred_all_nitrogen_avg$shuffled == "yes")),]

pred_all_nitrogen_avg$method[which((pred_all_nitrogen_avg$method == "rrBLUP_geno_microbiome") & (pred_all_nitrogen_avg$shuffled == "yes"))] = "rrBLUP_geno_shuffled_microbiome"



pred_all_nitrogen_avg$method <- factor(pred_all_nitrogen_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome", "rrBLUP_geno_shuffled_microbiome"), labels =c( "SNP", "SNP + Microbiome", "SNP + shuffled Microbiome"))



vis = as.character(data.frame(table(pred_all_nitrogen_avg$trait))$Var1)

i=2

my_comparisons <- list(c("SNP", "SNP + Microbiome"), c("SNP", "SNP + shuffled Microbiome"))


p_vari_shuffled_hn = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "HN")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = T, method.args = list(alternative = "less"))+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste("HN",vis[i],sep="_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


plist = list()

fsize = 16

for (i in 1:4) {
    plist[[i]] = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "HN")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = T, method.args = list(alternative = "less"))+
    xlab("") +
    ylab("") +
    ggtitle(paste("HN",vis[i],sep="_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
fig_hn <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]], ncol = 4, nrow = 1)
fig_hn






p_vari_shuffled_ln = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "LN")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = T, method.args = list(alternative = "less"))+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste("LN",vis[i],sep="_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


plist = list()

fsize = 16

for (i in 1:4) {
    plist[[i]] = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "LN")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = T, method.args = list(alternative = "less"))+
    xlab("") +
    ylab("") +
    ggtitle(paste("LN",vis[i],sep="_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
fig_ln <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]], ncol = 4, nrow = 1)
fig_ln

p_cobw_n_c <- ggarrange(p_cobw_shuffled, p_vari_shuffled_hn, p_vari_shuffled_ln, ncol = 3, nrow = 1)
p_cobw_n_c

p_cobw_n_c2 <- ggarrange(p_vari_shuffled_hn, p_vari_shuffled_ln, ncol = 2, nrow = 1)
p_cobw_n_c2

p_yieldc <- ggarrange(fig_hn, fig_ln, ncol = 1, nrow = 2)
p_yieldc


pdf("largedata/review/figures/p_cobw_n_c.pdf", width = 13, height= 6)
p_cobw_n_c
dev.off()

pdf("largedata/review/figures/p_cobw_n_c2.pdf", width = 9, height= 6)
p_cobw_n_c2
dev.off()

pdf("largedata/review/figures/p_yieldc_hn.pdf", width = 9, height= 10)
fig_hn
dev.off()

pdf("largedata/review/figures/p_yieldc_ln.pdf", width = 9, height= 10)
fig_ln
dev.off()


pdf("largedata/review/figures/p_yieldc.pdf", width = 18, height= 10)
p_yieldc
dev.off()

```
# mediation
```{r}
library(data.table)
library(dplyr)
library(tidyr)

sinfo_asv = fread("largedata/review/sinfo_asv.txt")
vis = fread("data/ppj220030-sup-0003-tables2.csv",header = T, data.table = F)

# vegetation indexes from column 7 to 14
id_trait = 12
# spread the combined data for trait "canopy coverage" with measurements from date 6-July to 5-Sep
vis_canopy = vis[,c(1:6,id_trait)]
table(vis_canopy$date)
vis_canopy = spread(vis_canopy, key = "date", value = colnames(vis)[id_trait]) 

# ceil_id : used for merging with corresponding phenotype
vis_canopy$ceil_id = paste("c",ceiling(vis_canopy$Row/2),sep = "_")
vis_canopy = vis_canopy[,c(18, 14, 6:13, 15:17)]
vis_canopy = vis_canopy[,c(1,3,7,13)]
sinfo_pheno_asv = merge(vis_canopy, sinfo_asv, by.x = "ceil_id", by.y = "ceil_id")




pall_s <- ggplot(sinfo_pheno_asv, aes(x=asv_000092, y=Aug12)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "bottom",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
pall_s

cor.test(sinfo_pheno_asv$asv_000092, sinfo_pheno_asv_isnp$Aug12, method = "spearman")




psn_s_aug12 <- ggplot(sinfo_pheno_asv, aes(x=asv_000092, y=Aug12, color=nitrogen)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_aug12

psn_s_aug22 <- ggplot(sinfo_pheno_asv, aes(x=asv_000092, y=Aug22, color=nitrogen)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_aug22

psn_s_sep5 <- ggplot(sinfo_pheno_asv, aes(x=asv_000092, y=Sept5, color=nitrogen)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_sep5


id_02_hn = which( (sinfo_pheno_asv$nitrogen %in% c("HN")))

cor.test(sinfo_pheno_asv$asv_000092[id_02_hn], sinfo_pheno_asv$Aug12[id_02_hn], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000092[id_02_hn], sinfo_pheno_asv$Aug22[id_02_hn], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000092[id_02_hn], sinfo_pheno_asv$Sept5[id_02_hn], method = "spearman")


id_02_ln = which( (sinfo_pheno_asv$nitrogen %in% c("LN")))

cor.test(sinfo_pheno_asv$asv_000092[id_02_ln], sinfo_pheno_asv$Aug12[id_02_ln], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000092[id_02_ln], sinfo_pheno_asv$Aug22[id_02_ln], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000092[id_02_ln], sinfo_pheno_asv$Sept5[id_02_ln], method = "spearman")




sinfo_pheno = gather(sinfo_pheno_asv, key = "date", value = "vari", 2:4)

id_02_hn = which((sinfo_pheno$nitrogen %in% c("HN"))) 

psn_s_hn <- ggplot(sinfo_pheno[id_02_hn,], aes(x=sinfo_pheno$asv_000092[id_02_hn], y=sinfo_pheno$vari[id_02_hn], color=sinfo_pheno$date[id_02_hn])) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_hn
psn_s_hn + scale_color_brewer(palette="Paired") 


id_02_ln = which( (sinfo_pheno$nitrogen %in% c("LN"))) 

psn_s_ln <- ggplot(sinfo_pheno[id_02_ln,], aes(x=sinfo_pheno$asv_000092[id_02_ln], y=sinfo_pheno$vari[id_02_ln], color=sinfo_pheno$date[id_02_ln])) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_ln

figure2 <- ggarrange( psn_s_hn,psn_s_ln, ncol = 1, nrow = 2)
figure2




id_02_hn = which((sinfo_pheno$nitrogen %in% c("HN"))) 

psn_s_hn <- ggplot(sinfo_pheno[id_02_hn,], aes(x=sinfo_pheno$asv_000524[id_02_hn], y=sinfo_pheno$vari[id_02_hn], color=sinfo_pheno$date[id_02_hn])) + 
  geom_point()+
  geom_smooth(method=lm) +
  xlab("") +
  ylab("") +
  ggtitle("") +
  theme_classic() +
  labs(fill = "") +
  #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_text(size=fsize, face="bold"),
        #axis.text.x = element_text(angle = 15, hjust=0.8),
        #legend.position = c(0.2, 0.8), 
        legend.position = "none",
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))
psn_s_hn

id_02_ln = which( (sinfo_pheno$nitrogen %in% c("LN"))) 

psn_s_ln <- ggplot(sinfo_pheno[id_02_ln,], aes(x=sinfo_pheno$asv_000524[id_02_ln], y=sinfo_pheno$vari[id_02_ln], color=sinfo_pheno$date[id_02_ln])) + 
  geom_point()+
  geom_smooth(method=lm) +
  xlab("") +
  ylab("") +
  ggtitle("") +
  theme_classic() +
  labs(fill = "") +
  #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_text(size=fsize, face="bold"),
        #axis.text.x = element_text(angle = 15, hjust=0.8),
        #legend.position = c(0.2, 0.8), 
        legend.position = "none",
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))
psn_s_ln

id_02_hn = which( (sinfo_pheno_asv$nitrogen %in% c("HN")))

cor.test(sinfo_pheno_asv$asv_000524[id_02_hn], sinfo_pheno_asv$Aug12[id_02_hn], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000524[id_02_hn], sinfo_pheno_asv$Aug22[id_02_hn], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000524[id_02_hn], sinfo_pheno_asv$Sept5[id_02_hn], method = "spearman")


id_02_ln = which( (sinfo_pheno_asv$nitrogen %in% c("LN")))

cor.test(sinfo_pheno_asv$asv_000524[id_02_ln], sinfo_pheno_asv$Aug12[id_02_ln], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000524[id_02_ln], sinfo_pheno_asv$Aug22[id_02_ln], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000524[id_02_ln], sinfo_pheno_asv$Sept5[id_02_ln], method = "spearman")



figure3 <- ggarrange( psn_s_hn,psn_s_ln, ncol = 1, nrow = 2)
figure3


figure = ggarrange( figure2, figure3, ncol = 2, nrow = 1)

pdf("largedata/review/figures/cor_vis_dates_asv.pdf", width = 10, height= 6)
figure
dev.off()

