---
title: "Genomic Selection practice"
author: "Zhikai Yang"
date: "Sept 9th, 2022"
output: NULL
---


## Normalize the path:

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE) 
knitr::opts_knit$set(root.dir=normalizePath('../../')) 
```



```{r}
library(data.table)
library(ggplot2)
library(agricolae)
library(ggpubr)
library(dplyr)

```



### mediation analysis

```{r}
library(data.table)
f <- list.files(path="largedata/review/med_output", pattern="mediators_.*3626asvs_spb_jyang", full.names = T)

fn <- list.files(path="largedata/review/med_output", pattern="mediators_.*_nitrogen.*3626asvs_spb_jyang", full.names = T)

f = f[-which(f %in% fn)]

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}


out$method  <- gsub(".*mediators_|_trait.*", "", out$file)
out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_3626asvs.*", "", out$file)

fwrite(out[,-7],"largedata/review/mediators_asvs_overall_3626asvs_spb.txt", sep="\t", row.names = F, quote = F)


```



```{r}
library(data.table)

f <- list.files(path="largedata/review/med_output", pattern="mediators_.*_nitrogen.*3626asvs_spb_jyang", full.names = T)


out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}


out$method  <- gsub(".*mediators_|_trait.*", "", out$file)
out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_nitrogen.*", "", out$file)
out$nitrogen = gsub(".*nitrogen_|_3626asvs.*","", out$file)

fwrite(out[,-7],"largedata/review/mediators_asvs_nitrogen_3626asvs_spb.txt", sep="\t", row.names = F, quote = F)


```

# overall
# including non-sig mediators
```{r}
mediators_asvs = fread("largedata/review/mediators_asvs_overall_3626asvs_spb.txt")


mediators_asvs_count_all = as.data.frame(table(mediators_asvs$id))

mediators_asvs_count_all = mediators_asvs_count_all[order(mediators_asvs_count_all$Freq, decreasing = T),]
rownames(mediators_asvs_count_all) = 1:nrow(mediators_asvs_count_all)

id_mediator_top1 = which(mediators_asvs_count_all$Var1 %in% unique(date_asv_mic_group_top1$asv[which(date_asv_mic_group_top1$top1p >0)]))

mediators_asvs_count_all_group = merge(mediators_asvs_count_all, asv_mic_group, by.x = "Var1", by.y = "ASV")


```

# just sig mediators after multiple comparison adjustment
```{r}
mediators_asvs = fread("largedata/review/mediators_asvs_overall_3626asvs_spb.txt")

mediators_asvs_sig = mediators_asvs[which(mediators_asvs$padj<=0.05),]

mediators_asvs_count_sig = as.data.frame(table(mediators_asvs_sig$id))



colnames(mediators_asvs_count_sig)[1] = "ASV"

asv_mic_group = fread("largedata/elife-75790-supp2-v2.csv", data.table = F)
asv_mic_group = asv_mic_group[,c(1,8)]

mediators_asvs_count_sig = merge(mediators_asvs_count_sig, asv_mic_group, by= "ASV")
mediators_asvs_count_sig = mediators_asvs_count_sig[order(mediators_asvs_count_sig$Freq, decreasing = T),]
rownames(mediators_asvs_count_sig) = 1:nrow(mediators_asvs_count_sig)

###


id_mediator_top1 = which(mediators_asvs_count_sig$Var1 %in% unique(date_asv_mic_group_top1$asv[which(date_asv_mic_group_top1$top1p >0)]))

mediators_asvs_count_sig_group = merge(mediators_asvs_count_sig, asv_mic_group, by.x = "Var1", by.y = "ASV")


mediators_asvs_sig_group = merge(asv_mic_group, mediators_asvs_sig,  by.x = "ASV", by.y = "id")

fwrite(mediators_asvs_sig_group, "data/supplementary/Supplemental_Table_S2_mediator_asvs.txt", sep="\t", row.names = F, quote = F)
```


#HN, LN
# just sig mediators after multiple comparison adjustment
```{r}
mediators_asvs_nitrogen = fread("largedata/review/mediators_asvs_nitrogen_3626asvs_spb.txt")

mediators_asvs_nitrogen_sig = mediators_asvs_nitrogen[which(mediators_asvs_nitrogen$padj<=0.05),]


mediators_asvs_ln_sig = mediators_asvs_nitrogen_sig[which(mediators_asvs_nitrogen_sig$nitrogen == "LN"),]
mediators_asvs_ln_count_sig = as.data.frame(table(mediators_asvs_ln_sig$id))
mediators_asvs_ln_count_sig = mediators_asvs_ln_count_sig[order(mediators_asvs_ln_count_sig$Freq, decreasing = T),]
rownames(mediators_asvs_ln_count_sig) = 1:nrow(mediators_asvs_ln_count_sig)

mediators_asvs_hn_sig = mediators_asvs_nitrogen_sig[which(mediators_asvs_nitrogen_sig$nitrogen == "HN"),]
mediators_asvs_hn_count_sig = as.data.frame(table(mediators_asvs_hn_sig$id))
mediators_asvs_hn_count_sig = mediators_asvs_hn_count_sig[order(mediators_asvs_hn_count_sig$Freq, decreasing = T),]
rownames(mediators_asvs_hn_count_sig) = 1:nrow(mediators_asvs_hn_count_sig)

```








### 3626 asvs + spb
### mediation analysis

```{r}
library(data.table)
f <- list.files(path="largedata/review/med_output", pattern="mediators_", full.names = T)

fn <- list.files(path="largedata/review/med_output", pattern="mediators_.*_nitrogen", full.names = T)

f = f[-which(f %in% fn)]

out <- data.frame()
for(i in 1:length(f)){
  tem <- read.csv(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}


out$method  <- gsub(".*mediators_|_trait.*", "", out$file)
out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_154tax.*", "", out$file)

fwrite(out[,-7],"largedata/review/mediators_asvs_overall.txt", sep="\t", row.names = F, quote = F)


```



```{r}
library(data.table)

f <- list.files(path="largedata/review/med_output", pattern="mediators_.*_nitrogen", full.names = T)


out <- data.frame()
for(i in 1:length(f)){
  tem <- read.csv(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}


out$method  <- gsub(".*mediators_|_trait.*", "", out$file)
out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_nitrogen.*", "", out$file)
out$nitrogen = gsub(".*nitrogen_|_154tax.*","", out$file)

fwrite(out[,-7],"largedata/review/mediators_asvs_nitrogen.txt", sep="\t", row.names = F, quote = F)


```

# overall
# including non-sig mediators
```{r}
mediators_asvs = fread("largedata/review/mediators_asvs_overall.txt")


mediators_asvs = mediators_asvs[-which(mediators_asvs$trait == "Canopy_Coverage"),]

mediators_asvs_count_all = as.data.frame(table(mediators_asvs$id))

mediators_asvs_count_all = mediators_asvs_count_all[order(mediators_asvs_count_all$Freq, decreasing = T),]
rownames(mediators_asvs_count_all) = 1:nrow(mediators_asvs_count_all)

id_mediator_top1 = which(mediators_asvs_count_all$Var1 %in% unique(date_asv_mic_group_top1$asv[which(date_asv_mic_group_top1$top1p >0)]))

mediators_asvs_count_all_group = merge(mediators_asvs_count_all, asv_mic_group, by.x = "Var1", by.y = "ASV")


```

# just sig mediators after multiple comparison adjustment
```{r}
mediators_asvs = fread("largedata/mediators_asvs.txt")

mediators_asvs_sig = mediators_asvs[which(mediators_asvs$padj<=0.05),]

mediators_asvs_count_sig = as.data.frame(table(mediators_asvs_sig$id))



mediators_asvs_count_sig = mediators_asvs_count_sig[order(mediators_asvs_count_sig$Freq, decreasing = T),]
rownames(mediators_asvs_count_sig) = 1:nrow(mediators_asvs_count_sig)

id_mediator_top1 = which(mediators_asvs_count_sig$Var1 %in% unique(date_asv_mic_group_top1$asv[which(date_asv_mic_group_top1$top1p >0)]))

mediators_asvs_count_sig_group = merge(mediators_asvs_count_sig, asv_mic_group, by.x = "Var1", by.y = "ASV")


mediators_asvs_sig_group = merge(asv_mic_group, mediators_asvs_sig,  by.x = "ASV", by.y = "id")

fwrite(mediators_asvs_sig_group[which(mediators_asvs_sig_group$method %in% c("fixed_bic", "fixed_eq")),], "data/supplementary/Supplemental_Table_S2_mediator_asvs.txt", sep="\t", row.names = F, quote = F)
```


#on hcc
```{r}
library(data.table)
library(dplyr)
library(tidyr)

sinfo_microbiome_geno = fread("largedata/Zhikai/sinfo_micobiome_geno.txt",header = T, data.table = F)


id_asv = which(colnames(sinfo_microbiome_geno) %in% c("asv_000092", "asv_000524"))

sinfo_asv = sinfo_microbiome_geno[,c(1:6,3633,id_asv)] # genotype (column 3633)
fwrite(sinfo_asv,"largedata/review/sinfo_asv.txt", sep="\t", row.names = F, quote = F)

```

```{r}
library(data.table)
library(dplyr)
library(tidyr)

sinfo_asv = fread("largedata/review/sinfo_asv.txt")
vis = fread("data/ppj220030-sup-0003-tables2.csv",header = T, data.table = F)

# vegetation indexes from column 7 to 14
id_trait = 12
# spread the combined data for trait "canopy coverage" with measurements from date 6-July to 5-Sep
vis_canopy = vis[,c(1:6,id_trait)]
table(vis_canopy$date)
vis_canopy = spread(vis_canopy, key = "date", value = colnames(vis)[id_trait]) 

# ceil_id : used for merging with corresponding phenotype
vis_canopy$ceil_id = paste("c",ceiling(vis_canopy$Row/2),sep = "_")
vis_canopy = vis_canopy[,c(18, 14, 6:13, 15:17)]
vis_canopy = vis_canopy[,c(1,3,7,13)]
sinfo_pheno_asv = merge(vis_canopy, sinfo_asv, by.x = "ceil_id", by.y = "ceil_id")




pall_s <- ggplot(sinfo_pheno_asv, aes(x=asv_000092, y=Aug12)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "bottom",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
pall_s

cor.test(sinfo_pheno_asv$asv_000092, sinfo_pheno_asv_isnp$Aug12, method = "spearman")




psn_s_aug12 <- ggplot(sinfo_pheno_asv, aes(x=asv_000092, y=Aug12, color=nitrogen)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_aug12

psn_s_aug22 <- ggplot(sinfo_pheno_asv, aes(x=asv_000092, y=Aug22, color=nitrogen)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_aug22

psn_s_sep5 <- ggplot(sinfo_pheno_asv, aes(x=asv_000092, y=Sept5, color=nitrogen)) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_sep5


id_02_hn = which( (sinfo_pheno_asv$nitrogen %in% c("HN")))

cor.test(sinfo_pheno_asv$asv_000092[id_02_hn], sinfo_pheno_asv$Aug12[id_02_hn], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000092[id_02_hn], sinfo_pheno_asv$Aug22[id_02_hn], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000092[id_02_hn], sinfo_pheno_asv$Sept5[id_02_hn], method = "spearman")


id_02_ln = which( (sinfo_pheno_asv$nitrogen %in% c("LN")))

cor.test(sinfo_pheno_asv$asv_000092[id_02_ln], sinfo_pheno_asv$Aug12[id_02_ln], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000092[id_02_ln], sinfo_pheno_asv$Aug22[id_02_ln], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000092[id_02_ln], sinfo_pheno_asv$Sept5[id_02_ln], method = "spearman")




sinfo_pheno = gather(sinfo_pheno_asv, key = "date", value = "vari", 2:4)

id_02_hn = which((sinfo_pheno$nitrogen %in% c("HN"))) 

psn_s_hn <- ggplot(sinfo_pheno[id_02_hn,], aes(x=sinfo_pheno$asv_000092[id_02_hn], y=sinfo_pheno$vari[id_02_hn], color=sinfo_pheno$date[id_02_hn])) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_hn
psn_s_hn + scale_color_brewer(palette="Paired") 


id_02_ln = which( (sinfo_pheno$nitrogen %in% c("LN"))) 

psn_s_ln <- ggplot(sinfo_pheno[id_02_ln,], aes(x=sinfo_pheno$asv_000092[id_02_ln], y=sinfo_pheno$vari[id_02_ln], color=sinfo_pheno$date[id_02_ln])) + 
    geom_point()+
    geom_smooth(method=lm) +
    xlab("") +
    ylab("") +
    ggtitle("") +
    theme_classic() +
    labs(fill = "") +
    #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          #axis.text.x = element_text(angle = 15, hjust=0.8),
          #legend.position = c(0.2, 0.8), 
          legend.position = "none",
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))
psn_s_ln

figure2 <- ggarrange( psn_s_hn,psn_s_ln, ncol = 1, nrow = 2)
figure2




id_02_hn = which((sinfo_pheno$nitrogen %in% c("HN"))) 

psn_s_hn <- ggplot(sinfo_pheno[id_02_hn,], aes(x=sinfo_pheno$asv_000524[id_02_hn], y=sinfo_pheno$vari[id_02_hn], color=sinfo_pheno$date[id_02_hn])) + 
  geom_point()+
  geom_smooth(method=lm) +
  xlab("") +
  ylab("") +
  ggtitle("") +
  theme_classic() +
  labs(fill = "") +
  #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_text(size=fsize, face="bold"),
        #axis.text.x = element_text(angle = 15, hjust=0.8),
        #legend.position = c(0.2, 0.8), 
        legend.position = "none",
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))
psn_s_hn

id_02_ln = which( (sinfo_pheno$nitrogen %in% c("LN"))) 

psn_s_ln <- ggplot(sinfo_pheno[id_02_ln,], aes(x=sinfo_pheno$asv_000524[id_02_ln], y=sinfo_pheno$vari[id_02_ln], color=sinfo_pheno$date[id_02_ln])) + 
  geom_point()+
  geom_smooth(method=lm) +
  xlab("") +
  ylab("") +
  ggtitle("") +
  theme_classic() +
  labs(fill = "") +
  #scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  theme(plot.title = element_text(size=20, face = "bold"), 
        axis.text=element_text(size=16, face="bold"),
        strip.text.x = element_text(size = 16, face = "bold"),
        axis.title=element_text(size=fsize, face="bold"),
        #axis.text.x = element_text(angle = 15, hjust=0.8),
        #legend.position = c(0.2, 0.8), 
        legend.position = "none",
        legend.title = element_text(size=fsize, face="bold"),
        legend.text = element_text(size=fsize))
psn_s_ln

id_02_hn = which( (sinfo_pheno_asv$nitrogen %in% c("HN")))

cor.test(sinfo_pheno_asv$asv_000524[id_02_hn], sinfo_pheno_asv$Aug12[id_02_hn], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000524[id_02_hn], sinfo_pheno_asv$Aug22[id_02_hn], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000524[id_02_hn], sinfo_pheno_asv$Sept5[id_02_hn], method = "spearman")


id_02_ln = which( (sinfo_pheno_asv$nitrogen %in% c("LN")))

cor.test(sinfo_pheno_asv$asv_000524[id_02_ln], sinfo_pheno_asv$Aug12[id_02_ln], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000524[id_02_ln], sinfo_pheno_asv$Aug22[id_02_ln], method = "spearman")
cor.test(sinfo_pheno_asv$asv_000524[id_02_ln], sinfo_pheno_asv$Sept5[id_02_ln], method = "spearman")



figure3 <- ggarrange( psn_s_hn,psn_s_ln, ncol = 1, nrow = 2)
figure3


figure = ggarrange( figure2, figure3, ncol = 2, nrow = 1)

pdf("largedata/review/figures/cor_vis_dates_asv.pdf", width = 10, height= 6)
figure
dev.off()

```
