---
title: "Genomic Selection practice"
author: "Zhikai Yang"
date: "March 9th, 2022"
output: NULL
---


## Normalize the path:

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE) 
knitr::opts_knit$set(root.dir=normalizePath('../../')) 
```




#on HCC cd /common/jyanglab/zhikaiyang/projects/GP_microbiome

```{r}
library(data.table)


f <- list.files(path="largedata/review/output_154Functional_Tax_Groupes", pattern="prediction_accuracy_seed.*_kinblup", full.names = T)

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
#out$date = gsub(".*date_|_shuffled.*", "", out$file)
out$shuffled  <- gsub(".*shuffled_|_kinblup.*", "", out$file)


fwrite(out[,-7], "largedata/review/pred_seeds_vis_shuffled_154Functional_Tax_Groupes_kinblup.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```

#on local

```{r}
library(data.table)
library(ggplot2)
library(agricolae)
library(ggpubr)
library(dplyr)

pred_all = fread("largedata/review/pred_seeds_vis_shuffled_154Functional_Tax_Groupes_kinblup.txt", header = T)


pred_all_avg =  as.data.frame( pred_all %>% group_by(trait, date, method, seed, shuffled) %>% summarise_all(mean))

pred_all_avg$date <- factor(pred_all_avg$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

pred_all_avg = pred_all_avg[-which((pred_all_avg$method == "rrBLUP") & (pred_all_avg$shuffled == "yes")),]

pred_all_avg$method[which((pred_all_avg$method == "rrBLUP_geno_microbiome") & (pred_all_avg$shuffled == "yes"))] = "rrBLUP_geno_shuffled_microbiome"



pred_all_avg$method <- factor(pred_all_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome", "rrBLUP_geno_shuffled_microbiome"), labels =c( "SNP", "SNP + ASV", "SNP + shuffled ASV by sample names"))



vis = as.character(data.frame(table(pred_all_avg$trait))$Var1)


my_comparisons <- list(c("SNP", "SNP + ASV"), c("SNP", "SNP + shuffled ASV by sample names"))





plist = list()

fsize = 16

for (i in 1:8) {
    plist[[i]] = ggboxplot(pred_all_avg[which(pred_all_avg$trait == vis[i]),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    #stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(vis[i]) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(-0.01, 1)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
figure <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]],plist[[5]],plist[[6]],plist[[7]],plist[[8]], ncol = 2, nrow = 4)
figure


i=6


p_vari_shuffled_aug12 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "bottom", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


p_vari_shuffled_aug22 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "bottom", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


p_vari_shuffled_sep5 = ggboxplot(pred_all_avg[which((pred_all_avg$trait == vis[i]) & (pred_all_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "bottom", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))



```




#coef of microbiome, on HCC cd /common/jyanglab/zhikaiyang/projects/GP_microbiome
```{r}
library(data.table)

f <- list.files(path="largedata/review/output_154Functional_Tax_Groupes", pattern="coef_mic_trait", full.names = T)

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_jyang.*", "", out$file)
out = out[,-3]



fwrite(out, "largedata/review/coef_mic_all_traits_dates_154tax.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```

#coef of microbiome spread
```{r}

coef_all = fread("largedata/review/coef_mic_all_traits_dates_154tax.txt", data.table = F)


coef_all$date <- factor(coef_all$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))


vis = as.character(data.frame(table(coef_all$trait))$Var1)

plist = list()

for (i in 1:8) {
    plist[[i]] = ggboxplot(coef_all[which((coef_all$trait == vis[i])),], x = "date", y = "coef", color = "date", palette = "jco",add = "jitter") +
    #stat_compare_means(aes(group = method), label = "p.signif", paired = T)+
    xlab("") +
    ylab("Coef of ASVs ") +
    ggtitle(vis[i]) +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
figure <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]],plist[[5]],plist[[6]],plist[[7]],plist[[8]], ncol = 2, nrow = 4)
figure

```



#separate HN and LN, on HCC
```{r}
library(data.table)
f <- list.files(path="largedata/review/output_154Functional_Tax_Groupes", pattern="prediction_accuracy_seed.*_nitrogen_.*_kinblup", full.names = T)

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$nitrogen = gsub(".*nitrogen_|_shuffled.*", "", out$file)
out$shuffled  <- gsub(".*shuffled_|_kinblup.*", "", out$file)

fwrite(out[,-7], "largedata/review/pred_seeds_nitrogen_vis_shuffled_names_154tax_kinblup.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```


#separate HN and LN, all vegetation indexes visualization 
```{r}
pred_all_nitrogen = fread("largedata/review/pred_seeds_nitrogen_vis_shuffled_names_154tax_kinblup.txt", data.table = F)

pred_all_nitrogen_avg =  as.data.frame( pred_all_nitrogen %>% group_by(trait, date, method, seed, nitrogen, shuffled) %>% summarise_all(mean))

pred_all_nitrogen_avg$date <- factor(pred_all_nitrogen_avg$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

pred_all_nitrogen_avg = pred_all_nitrogen_avg[-which((pred_all_nitrogen_avg$method == "rrBLUP") & (pred_all_nitrogen_avg$shuffled == "yes")),]

pred_all_nitrogen_avg$method[which((pred_all_nitrogen_avg$method == "rrBLUP_geno_microbiome") & (pred_all_nitrogen_avg$shuffled == "yes"))] = "rrBLUP_geno_shuffled_microbiome"



pred_all_nitrogen_avg$method <- factor(pred_all_nitrogen_avg$method, levels = c( "rrBLUP", "rrBLUP_geno_microbiome", "rrBLUP_geno_shuffled_microbiome"), labels =c( "SNP", "SNP + ASV", "SNP + shuffled ASV by sample names"))



vis = as.character(data.frame(table(pred_all_nitrogen_avg$trait))$Var1)

i=6

my_comparisons <- list(c("SNP", "SNP + ASV"), c("SNP", "SNP + shuffled ASV by sample names"))


p_vari_shuffled_hn = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "HN")),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    #stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle("HN") +
    theme_classic() +
    labs(fill = "") +
    scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_ln = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i]) & (pred_all_nitrogen_avg$nitrogen == "LN")),], x = "date", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    #stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle("LN") +
    theme_classic() +
    labs(fill = "") +
    scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_n_c <- ggarrange(p_vari_shuffled_hn, p_vari_shuffled_ln, ncol = 2, nrow = 1)
p_vari_n_c



pdf("largedata/p_vari_n_c.pdf", width = 10, height= 6)
p_vari_n_c
dev.off()


p_vari_shuffled_hn_aug12 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "HN") & (pred_all_nitrogen_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_hn_aug22 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "HN") & (pred_all_nitrogen_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_hn_sep5 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "HN") & (pred_all_nitrogen_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_blank(),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))


p_vari_shuffled_ln_aug12 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "LN") & (pred_all_nitrogen_avg$date == "Aug-12")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-12", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_ln_aug22 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "LN") & (pred_all_nitrogen_avg$date == "Aug-22")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Aug-22", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_shuffled_ln_sep5 = ggboxplot(pred_all_nitrogen_avg[which((pred_all_nitrogen_avg$trait == vis[i])& (pred_all_nitrogen_avg$nitrogen == "LN") & (pred_all_nitrogen_avg$date == "Sep-5")),], x = "method", y = "accuracy", color = "method", palette = "jco",add = "jitter") +
    stat_compare_means(comparisons = my_comparisons, label = "p.signif", method = "t.test", aes(group = method), paired = F)+
    xlab("") +
    ylab("Prediction Accuracy ") +
    ggtitle(paste(vis[i], "Sep-5", sep = "_")) +
    theme_classic() +
    labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_blank(),
          axis.text.x = element_blank(),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_vari_hn_c <- ggarrange(p_vari_shuffled_hn_aug12, p_vari_shuffled_hn_aug22, p_vari_shuffled_hn_sep5, ncol = 1, nrow = 3)
p_vari_hn_c

p_vari_ln_c <- ggarrange(p_vari_shuffled_ln_aug12, p_vari_shuffled_ln_aug22, p_vari_shuffled_ln_sep5, ncol = 1, nrow = 3)
p_vari_ln_c


```




#coef of microbiome, on HCC cd /common/jyanglab/zhikaiyang/projects/GP_microbiome
```{r}
library(data.table)

f <- list.files(path="largedata/review/output_154Functional_Tax_Groupes", pattern="coef_mic_trait", full.names = T)

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_jyang.*", "", out$file)
out = out[,-3]



fwrite(out, "largedata/review/coef_mic_all_traits_dates_154tax.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```

#coef of microbiome, on HCC cd /common/jyanglab/zhikaiyang/projects/GP_microbiome
```{r}
library(data.table)

f <- list.files(path="largedata/review/output_154Functional_Tax_Groupes", pattern="coef_mic_trait.*_nitrogen_", full.names = T)

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_nitrogen.*", "", out$file)
out$nitrogen = gsub(".*nitrogen_|_jyang.*", "", out$file)

out = out[,-3]



fwrite(out, "largedata/review/coef_mic_all_traits_dates_nitrogen_154tax.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```


#top abs_scaled asvs
```{r}
coef_vis = fread("largedata/review/coef_mic_all_traits_dates_nitrogen_154tax.txt", data.table = F)

coef_vis_hn = coef_vis[which(coef_vis$nitrogen == "HN"),]
out <- data.frame()
for(i in 1:length(vis)){
  for (j in 1:length(date)) {
    tem <- coef_vis_hn[which((coef_vis_hn$trait == vis[i])  & (coef_vis_hn$date == date[j])),]
    tem$abs_coef = abs(tem$coef)
    tem = tem[order(-tem$abs_coef),]
    tem$rank = 1:nrow(tem)
    tem$top1p = ifelse(tem$rank <= 15, 1, 0)
    tem$top100 = ifelse(tem$rank <= 100, 1, 0)
    out <- rbind(out, tem)
  }
}

asv_coef_vis_hn = as.data.frame(out[,-(3:5)] %>% group_by(asv) %>% summarise_all(sum))
asv_date_coef_vis_hn = as.data.frame(out[,-c(3,5)] %>% group_by(date,asv) %>% summarise_all(sum))




coef_vis_ln = coef_vis[which(coef_vis$nitrogen == "LN"),]
out <- data.frame()
for(i in 1:length(vis)){
  for (j in 1:length(date)) {
    tem <- coef_vis_ln[which((coef_vis_ln$trait == vis[i])  & (coef_vis_ln$date == date[j])),]
    tem$abs_coef = abs(tem$coef)
    tem = tem[order(-tem$abs_coef),]
    tem$rank = 1:nrow(tem)
    tem$top1p = ifelse(tem$rank <= 15, 1, 0)
    tem$top100 = ifelse(tem$rank <= 100, 1, 0)
    out <- rbind(out, tem)
  }
}

asv_coef_vis_ln = as.data.frame(out[,-(3:5)] %>% group_by(asv) %>% summarise_all(sum))
asv_date_coef_vis_ln = as.data.frame(out[,-c(3,5)] %>% group_by(date,asv) %>% summarise_all(sum))



#asv_date_coef_vis_ln = asv_date_coef_vis_ln[which(asv_date_coef_vis_ln$top100 > 0),]
asv_date_coef_vis_ln$top1p = as.character(asv_date_coef_vis_ln$top1p)
asv_date_coef_vis_ln$top100 = as.character(asv_date_coef_vis_ln$top100)
asv_date_coef_vis_ln$count = 1
date_ntrait_count_top1p = as.data.frame(asv_date_coef_vis_ln[,c(1,6,8)] %>% group_by(date,top1p) %>% summarise_all(sum))
date_ntrait_count_top1p$date <- factor(date_ntrait_count_top1p$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

p_date_ntrait_count_top1p = ggplot(date_ntrait_count_top1p[-which(date_ntrait_count_top1p$top1p == "0"),], aes(x=date, y=count, fill=top1p)) +
  geom_bar(stat="identity", position=position_dodge())+
  scale_fill_brewer(palette="Blues")+
    xlab("") +
    ylab("Count of top 1% ASVs") +
    ggtitle("Histograms of Num of Traits by different dates ") +
    theme_dark() +
    #labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=14, face="bold"),
          strip.text.x = element_text(size = 14, face = "bold"),
          axis.title=element_text(size=fsize-2, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "right", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_date_ntrait_count_top1p


date_asv_top1 = asv_date_coef_vis_ln[,c(1,2,6)]
asv_mic_group = fread("largedata/elife-75790-supp2-v2.csv", data.table = F)
asv_mic_group = asv_mic_group[,c(1,8)]
date_asv_mic_group_top1 = merge(date_asv_top1, asv_mic_group, by.x = "asv", by.y = "ASV")
mic_group_anno = fread("largedata/elife-75790-supp3-v2.csv", data.table = F)

count_mic_g = as.data.frame(table(date_asv_mic_group_top1$Functional_Tax_Group))
count_mic_g$Var1[which(!(count_mic_g$Var1 %in% unique(mic_group_anno$Microbial_Group)))]

#HN
date_asv_mic_group_top1_anno_HN = merge(date_asv_mic_group_top1, mic_group_anno[which(mic_group_anno$nitrogen_treatment == "HN"),], by.x = "Functional_Tax_Group", by.y ="Microbial_Group")

date_asv_mic_group_top1_anno_HN$iftop1p = ifelse(date_asv_mic_group_top1_anno_HN$top1p >= 1, "Top 1% ASVs", "Not Top 1% ASVs")

date_asv_mic_group_top1_anno_HN$date <- factor(date_asv_mic_group_top1_anno_HN$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

#LN
date_asv_mic_group_top1_anno_LN = merge(date_asv_mic_group_top1, mic_group_anno[which(mic_group_anno$nitrogen_treatment == "LN"),], by.x = "Functional_Tax_Group", by.y ="Microbial_Group")

date_asv_mic_group_top1_anno_LN$iftop1p = ifelse(date_asv_mic_group_top1_anno_LN$top1p >= 1, "Top 1% ASVs", "Not Top 1% ASVs")

date_asv_mic_group_top1_anno_LN$date <- factor(date_asv_mic_group_top1_anno_LN$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))

date_asv_mic_group_jtop1_anno_LN_aug12 = date_asv_mic_group_top1_anno_LN[which((date_asv_mic_group_top1_anno_LN$date == "Aug-12") & (date_asv_mic_group_top1_anno_LN$iftop1p == "Top 1% ASVs")), ]

fwrite(date_asv_mic_group_jtop1_anno_LN_aug12, "largedata/date_asv_mic_group_jtop1_anno_LN_aug12.txt", sep = "\t", quote = F, row.names = F, col.names = T)

date_asv_mic_group_jtop1_anno_LN_aug12 = fread("largedata/date_asv_mic_group_jtop1_anno_LN_aug12.txt", data.table = F)

p_h2_HN = ggboxplot(date_asv_mic_group_top1_anno_HN, x = "date", y = "Heritability", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F)+
    xlab("") +
    ylab("Heritability") +
    ggtitle("HN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_h2_HN

p_h2_LN = ggboxplot(date_asv_mic_group_top1_anno_LN, x = "date", y = "Heritability", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F)+
    xlab("") +
    ylab("Heritability") +
    ggtitle("LN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_h2_LN


p_Selection_coeff_HN = ggboxplot(date_asv_mic_group_top1_anno_HN, x = "date", y = "Selection_coeff", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F)+
    xlab("") +
    ylab("Selection_coeff") +
    ggtitle("HN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_Selection_coeff_HN

p_Selection_coeff_LN = ggboxplot(date_asv_mic_group_top1_anno_LN, x = "date", y = "Selection_coeff", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F)+
    xlab("") +
    ylab("Selection_coeff") +
    ggtitle("LN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_Selection_coeff_LN

library(ggpubr)
p_anno <- ggarrange(p_h2_HN, p_h2_LN, p_Selection_coeff_HN, p_Selection_coeff_LN, ncol = 2, nrow = 2)
p_anno

pdf("largedata/p_anno.pdf", width = 12, height= 12)
p_anno
dev.off()

p_Selection_coeff_LN = ggboxplot(date_asv_mic_group_top1_anno_LN, x = "date", y = "Selection_coeff", color = "iftop1p", palette = "jco",add = "jitter") +
    stat_compare_means(aes(group = iftop1p), label = "p.signif", paired = F)+
    xlab("") +
    ylab("Selection_coeff") +
    ggtitle("LN") +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 0, hjust=0.8),
          legend.position = "bottom", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

p_Selection_coeff_LN
pdf("largedata/p_Selection_coeff_LN.pdf", width = 12, height= 12)
p_Selection_coeff_LN
dev.off()

```


#coef of SNPs, on HCC
```{r}
f <- list.files(path="largedata/outputs", pattern="coef_snps_trait", full.names = T)

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_jyang.*", "", out$file)
out = out[,-3]



fwrite(out, "largedata/coef_snps_all_traits_dates.txt", sep = "\t", quote = F, row.names = F, col.names = T)



```

#coef of snps
```{r}

coef_all_snps = fread("largedata/coef_snps_all_traits_dates.txt", data.table = F)

coef_all_snps$date <- factor(coef_all_snps$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))


vis = as.character(data.frame(table(coef_all_snps$trait))$Var1)

plist = list()

for (i in 1:9) {
    plist[[i]] = ggboxplot(coef_all_snps[which((coef_all_snps$trait == vis[i])),], x = "date", y = "coef", color = "date", palette = "jco",add = "jitter") +
    #stat_compare_means(aes(group = method), label = "p.signif", paired = T)+
    xlab("") +
    ylab("Coef of SNPs ") +
    ggtitle(vis[i]) +
    theme_classic() +
    labs(fill = "") +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.x = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=fsize, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "none", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
figure <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]],plist[[5]],plist[[6]],plist[[7]],plist[[8]],plist[[9]], ncol = 3, nrow = 3)
figure

pdf("largedata/coef_snps.pdf", width = 12, height= 9)
figure
dev.off()


```


# R2 of different sources
```{r}
f <- list.files(path="largedata/outputs", pattern="r2_d_rrblup_geno_microbiome_trait", full.names = T)

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_jyang.*", "", out$file)

out = out[,-4]
fwrite(out, "largedata/r2_d_rrblup_geno_microbiome_traits.txt", sep = "\t", quote = F, row.names = F, col.names = T)

```



# visualization of R2 of different sources
```{r}
r2_d_rrblup_geno_microbiome = fread("largedata/r2_d_rrblup_geno_microbiome_traits.txt", data.table = F)



fsize =16


r2_d_rrblup_geno_microbiome$source <- factor(r2_d_rrblup_geno_microbiome$source, levels = c("total", "fixed_and_random", "fixed", "random", "genome", "microbiome", "residual"), labels =c("total", "fixed_and_random", "fixed", "random", "genome", "microbiome", "residual"))

r2_d_rrblup_geno_microbiome$date <- factor(r2_d_rrblup_geno_microbiome$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))


vis = as.character(data.frame(table(r2_d_rrblup_geno_microbiome$trait))$Var1)

plist = list()

for (i in 1:9) {
    plist[[i]] = ggplot(r2_d_rrblup_geno_microbiome[which((r2_d_rrblup_geno_microbiome$trait == vis[i])),], aes(x=date, y=r2, fill=source)) +
  geom_bar(stat="identity", position=position_dodge())+
  scale_fill_brewer(palette="Paired")+
    xlab("") +
    ylab("R-squared") +
    ggtitle(vis[i]) +
    theme_classic() +
    #labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=14, face="bold"),
          strip.text.x = element_text(size = 14, face = "bold"),
          axis.title=element_text(size=fsize-2, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "right", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
figure <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]],plist[[5]],plist[[6]],plist[[7]],plist[[8]],plist[[9]], ncol = 3, nrow = 3)
figure



```



# R2 of different sources
```{r}
f <- list.files(path="largedata/outputs", pattern="r2_d_nitrogen_rrblup_geno_microbiome_trait", full.names = T)

out <- data.frame()
for(i in 1:length(f)){
  tem <- fread(f[i], header=TRUE)
  tem$file <- f[i]
  out <- rbind(out, tem)
}

out$trait  <- gsub(".*trait_|_date.*", "", out$file)
out$date  <- gsub(".*date_|_jyang.*", "", out$file)

out = out[,-4]
fwrite(out, "largedata/r2_d_nitrogen_rrblup_geno_microbiome_traits.txt", sep = "\t", quote = F, row.names = F, col.names = T)

```



# visualization of R2 of different sources
```{r}
r2_d_rrblup_geno_microbiome = fread("largedata/r2_d_nitrogen_rrblup_geno_microbiome_traits.txt", data.table = F)

r2_d_rrblup_geno_microbiome = r2_d_rrblup_geno_microbiome[which(r2_d_rrblup_geno_microbiome$source %in% c("nitrogen", "genome", "microbiome")),]

fsize =16


r2_d_rrblup_geno_microbiome$source <- factor(r2_d_rrblup_geno_microbiome$source, levels = c("nitrogen", "genome", "microbiome"), labels =c("nitrogen", "genome", "microbiome"))

r2_d_rrblup_geno_microbiome$date <- factor(r2_d_rrblup_geno_microbiome$date, levels = c( "Aug12", "Aug22", "Sept5"), labels =c( "Aug-12", "Aug-22", "Sep-5"))


vis = as.character(data.frame(table(r2_d_rrblup_geno_microbiome$trait))$Var1)

plist = list()

for (i in 1:9) {
    plist[[i]] = ggplot(r2_d_rrblup_geno_microbiome[which((r2_d_rrblup_geno_microbiome$trait == vis[i])),], aes(x=date, y=r2, fill=source)) +
  geom_bar(stat="identity", position=position_dodge())+
  scale_fill_brewer(palette="Paired")+
    xlab("") +
    ylab("Variance Explained") +
    ggtitle(vis[i]) +
    theme_classic() +
    #labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=14, face="bold"),
          strip.text.x = element_text(size = 14, face = "bold"),
          axis.title=element_text(size=fsize-2, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "right", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

    
}


library(ggpubr)
figure <- ggarrange(plist[[1]], plist[[2]],plist[[3]],plist[[4]],plist[[5]],plist[[6]],plist[[7]],plist[[8]],plist[[9]], ncol = 3, nrow = 3)
figure

p_vari_var_part = ggplot(r2_d_rrblup_geno_microbiome[which((r2_d_rrblup_geno_microbiome$trait == vis[6])),], aes(x=date, y=r2, fill=source)) +
  geom_bar(stat="identity", position=position_dodge())+
  scale_fill_brewer(palette="Paired")+
    xlab("Days after microbiome sampling") +
    ylab("Variance Explained") +
    ggtitle(vis[6]) +
    theme_classic() +
    #labs(fill = "") +
    #scale_y_continuous(limits = c(0.15, 0.8)) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=14, face="bold"),
          strip.text.x = element_text(size = 14, face = "bold"),
          axis.title=element_text(size=fsize-2, face="bold"),
          axis.text.x = element_text(angle = 25, hjust=0.8),
          legend.position = "right", 
          legend.title = element_text(size=fsize, face="bold"),
          legend.text = element_text(size=fsize))

pdf("largedata/p_vari_var_partition.pdf", width = 6, height= 6)
p_vari_var_part
dev.off()



```

